# Differential gene analysis {#Differential}

Once you obtain gene counts from the preprocessing modules of RIMA, you can compare gene expression levels for different sample phenotypes.  For example, you may wish to know which genes are differentially expressed between patients who respond and do not respond to a particular treatment.  In addition, you may wish to know if particular gene sets are enriched in expression between different phenotypes. These gene sets may be associated with a known biological pathway, process or molecular function.   

The differential gene analysis module of RIMA allows you to perform these comparisons.


## 1. Differential gene expression using DESeq2

The <a href='https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8'>DESeq2</a> is the most commonly used package that provides methods for comparison of gene expression levels.  The package is described in:

    Love, M.I., Huber, W., Anders, S. (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550. 10.1186/s13059-014-0550-8

Details about how to run DeSeq2 are also included in the DESeq2 
[vignette](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).

### Inputs of DESeq2
The DESeq2 accepts four kinds of inputs: 1) Transcript abundance; 2) Read count matrix: 3) htseq-count files and 4) A SummarizedExpreriment object. We processed our gene quantification using Salmon in our preprocess module, so we will run DESeq2 using the output in the quant.sf files provided from salmon. The quant.sf files were generated during the preprocessing module.   [vignette](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).

We first created a tx2gene.csv files that contains transcripts id and gene names as below:
```{r eval=FALSE}
ID,GENENAME
ENST00000000233.8,ARF5
ENST00000000412.6,M6PR
ENST00000000442.9,ESRRA
ENST00000001008.5,FKBP4
ENST00000001146.5,CYP26B1
ENST00000002125.7,NDUFAF7
ENST00000002165.9,FUCA2
ENST00000002501.9,DBNDD1
ENST00000002596.5,HS3ST1
```
Then we uses tximport to import transcripts abundance data from the quant.sf files from salmon as described in the DESeq2:

```{r eval=FALSE}
### filelist are a list of quant.sf files generated by Salmon gene quantification.
txi <- tximport(filelist, type=Salmon, tx2gene=tx2gene)
```
## Different gene expression analysis
After loading the transcripts abundance, we are ready to find out differentially expressed gene between your selected condition. We need to be careful when running DESeq2 with batch effects.
### Running DESeq2 without batch effect
We will run DESeq2 to determine differential expression for a particular phenotype.  In our case, we are interested in the genes differential between responders and non-responders, which is the "Responders" column headings in the metasheet.
We  uses the information in "designs" for DESeq2 in the following manner:

```
ddsTxi <- DESeqDataSetFromTximport(txi,
                                       colData = samples,
                                       design = ~ Responder)

dds <- DESeq(ddsTxi)
```

### Running DESeq2 with batch effect
DESeq2 can account for any batch effect by providing the batch variable as a covariate. If your batch effect analysis from the preprocessing module indicated that there is a batch effect in your samples, set the "batch" field in config.yaml to the appropriate column name in your metasheet.
We uses the information in "batch" for DESeq2 in the following manner:

```
ddsTxi <- DESeqDataSetFromTximport(txi,
                                       colData = samples,
                                       design = ~ syn_batch + Responder)
dds <- DESeq(ddsTxi)
```


### Output from DESeq2

Output file of DESeq2 generated values for baseMean, log2FoldChange, lfcSE, stat, pvalue, padj. These fields have the following meaning:

  - baseMean = the average of the normalized counts for all samples
  - Log2FoldChange = log base 2 fold changes for the condition tested.  For example if the file is Responder_nonresponder_VS_resonder_DESqe2_raw.txt, Log2FoldChange will be the estimated logarithmic fold change log2(nonresponder/responder).
  - lfcSE = Log2FoldChange Standard Error
  - stat= Wald statistic
  - pvalue = Wald test p value for the Log2FoldChange estimate
  - padj. = FDR adjusted p value (Benjamini-Hochberg adjusted p value)

{comparison condition}_DEseq2_raw.txt lists Gene names directly from the DESeq2 output.  It also contains a value for -log10(padj).

### Volcano Plot(s)
Vocalno plot is usually used to visualize the results of differentially expressed genes with x axis represents the logFC and y axis represent the -log10(P.value). Up-regulated genes has positive logFC and down-regulated genes has negative logFC. \
Below is an example of volcano plot from previous differential results and we labeled the genes with below cutoff:

  - FDR < 0.05
  - Log2FoldChange > 1

```{r fig.align='center', echo=FALSE, fig.cap='Volcano plot'}
knitr::include_graphics('images/volcano.png', dpi = NA)
```


## Gene set enrichment analysis (GSEA)

GSEA was developed by UC San Diego and the Broad Institute.  It is described in

    Subramanian, Tamayo, et al. (2005, PNAS 102, 15545-15550) and Mootha, Lindgren, et al. (2003, Nat Genet 34, 267-273).
  
GSEA tools and software can be found on the GSEA [website](https://www.gsea-msigdb.org/gsea/index.jsp).

Gene Set Enrichment Analysis can be used to detect patterns in differential expression that affect particular gene pathways, molecular functions, cellular components or processes.  We uses the GSEA software to look for enrichment in the following gene sets:

  - KEGG canonical pathways
  - Gene Ontology sets -- including Biological Process (BP), Cellular Component (CC) and Molecular Function (MF).

RIMA uses a ranked list of DESeq2 outputs.  Ranking is based on the following:

```
sign(data$log2FoldChange) * (-log10(data$pvalue))
```

GSEA output is reported in 4 text files:

  - {comparison condition_phenotype1_VS_phenotype2}_KEGG_terms.txt
  - {comparison condition_phenotype1_VS_phenotype2}_GO_CC_terms.txt
  - {comparison condition_phenotype1_VS_phenotype2}_GO_BP_terms.txt
  - {comparison condition_phenotype1_VS_phenotype2}_GO_MF_terms.txt
  
Each of these tab separated files contains the following information:

  - ID = ID of the gene set
  - Description = description of the gene set
  - setSize = number of genes in the gene set after filtering out genes not in the expression data.
  - enrichmentScore = the degree to which the gene set is enriched in the top or bottom of the ranked set of genes.
  - NES = normalized gene enrichment score.  
  - pvalue = nominal p value
  - p.adjust = family-wise error adjusted p value
  - qvalues = FDR adjusted p value
  - rank = the position in the ranked list at which the maximal enrichment score was achieved.
  - leading_edge = 3 statistics (tags, list and signal) on the subset of genes that most influence the enrichment score
  - core_enrichment group = the subset of genes that contribute most to the enrichment score.
  - significance = a representation of the p.adjust value using stars.  One star is set for p.adjust <0.1, two stars <0.5 and three stars < 0.01.

Please see the [GSEA documentation](https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html) for more information about these values.


## Single sample gene set enrichment analysis (ssGSEA)

ssGSEA is a different method of trying to find enriched pathways in your dataset.  GSEA uses the differential expression patterns of all samples to generate enrichment scores based on phenotypic differences.  In contrast, ssGSEA generates enrichment scores for each sample.  Phenotypic differences can then be used to compare enriched pathways rather than differential genes expression levels.  Please see [ssGSEA](https://gsea-msigdb.github.io/ssGSEA-gpmodule/v10/index.html) for more information.

The input to ssGSEA is the tpm_convertID.batch file generated by LIMMA.  The file contains log2 transformed tpm values.

config.yaml setting for ssGSEA...

output from ssGSEA....

