# Immune Infiltration {#Infiltration}

**Cell infiltration estimation in the tumor microenvironment**\
The tumor microenviroment (**TME**) is a complicated system consisting of immune cells, stromal cells, and extracellular factors. Estimating cell components is essential for classifying the distinct tumor immune microenvironment (TIME) phenotype. Furthermore, dissecting the TIME by evaluating cell components plays a significant role in untangling the mechanism of tumor progression and immune evasion, which is essential for providing a guideline for immunotherapy selection. 

There are two major approaches for estimating cell infiltration in the TME --  1) deconvolution-based and 2) marker-based approaches. Deconvolution-based methods use weighted mathematical models that infer cell type fractions from the bulk gene expression of a given list of signature genes.  Algorithms such as TIMER, quanTIseq, EPIC, and CIBERSORT use deconvolution-based methods.  Algorithms such as xCell and MCP-counter use marker-based approaches to quantify the enrichment score utilizing a list of cell-specific marker genes. <a href="https://github.com/icbi-lab/immunedeconv">ImmuneDeconv</a> is an R package that integrates these six algorithms. All six algorithms estimate cell infiltration of immune cells and stromal cells, while EPIC and quanTIseq also assess uncharacterized cells defined as cancer cells. Of note, CIBERSORT absolute mode, EPIC, and quanTIseq support inter-comparison between sample groups and intra-comparison between cell types. And TIMER, xCell, and MCP-counter support only inter-comparison between sample groups for one cell type. Immune cell deconvolution is still an open question in the field, and different algorithms could give slightly different results. Each algorithm estimates one general cell type or specific cell subtype. When a gold standard is not available, is is best to focus on the observations consistently predicted by the various algorithms. \

## Cibersort
<a href="https://www.nature.com/articles/nmeth.3337">CIBERSORT</a> is a deconvolution-based method that uses the v-support vector regression method (v-SVR) to estimate each immune cell type's relative proportion from a gene expression profile (GEP). It requires an input of reference gene expression signatures. CIBERSORT provides a gene signature matrix (LM22.txt) which contains 547 genes and distinguishes 7 T-cell types, na√Øve and memory B cells, plasma cells, and NK cells. The CIBERSORT absolute mode scales cellular fractions to a score that reflects each cell type's absolute proportion, allowing comparison across both samples and cell types.
The **CIBERSORT source code** and **LM22 signature matrix** are only freely available to academic users -- register for access on the <a href="https://cibersort.stanford.edu/">CIBERSORT website </a>. \

To run CIBERSORT using ImmuneDeconv:
```{r, eval = TRUE}
#demo
suppressMessages(library(immunedeconv))
#read gene expression matrix 
input <- "tpm_convertID.txt"

#load CIBERSORT source code and signiture matrix 
source("CIBERSORT.R")
sig_matrix <- "LM22.txt"

#Run CIBERSORT abs 
#The number of permutation
cibersort_perm = 100
#Quantile normalization of input mixture, default = FALSE for RNA-Seq data
cibersort_qn = FALSE
#whether to apply absolute mode in cibersort
cibersort_abs = TRUE
#sig.score = for each mixture sample, define S as the median expression,level of all genes in the signature matrix divided by the median expression level of all genes in the mixture. Multiple cell subset fractions by S.
cibersort_abs_method = "sig.score"
res_ciber <- CIBERSORT(sig_matrix, input, perm = cibersort_perm, QN = cibersort_qn, absolute = cibersort_abs,
                       abs_method = cibersort_abs_method)

head(res_ciber,3)
```

## TIMER
<a href="http://cistrome.org/TIMER/">TIMER</a> (Tumor IMmune Estimation Resource) uses linear least square regression to estimate six tumor-infiltrating immune cell types (B, CD4T, CD8T, neutrophils, macrophages, and dendritic cells). Since the tumor purity is very different across different cancer cell types, TIMER selects specific genes as immune signatures based on the correlation between gene expression and tumor purity. (see <a href="https://pubmed.ncbi.nlm.nih.gov/27549193/">Bo Li, et al.</a> for more information.\

Tutorial of TIMER:\
<iframe width="560" height="315" src="https://www.youtube.com/embed/94v8XboCrXU" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe> \

Tutorial of TIMER2:\

<iframe width="560" height="315" src="https://www.youtube.com/embed/2hmxioq1pJo" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe> \

Note that the TIMER score can only be compared within samples, and the current version of TIMER supports 23 different TCGA cancer types.
```{r,eval=FALSE}

#load gene expression matrix 
df <- read.table(input, sep = ",", row.names = 1, header = TRUE, check.names = FALSE)

#The TCGA cancer acronyms of demo data
cancertype <- "GBM"
#Available cancer type 
immunedeconv::timer_available_cancers
res_timer = as.data.frame(deconvolute(df, "timer",indications=rep(tolower(cancertype),ncol(df))))
```

## quanTIseq
<a href="https://icbi.i-med.ac.at/software/quantiseq/doc/index.html">quanTIseq</a> uses constrained least square regression to estimate absolute proportions of the infiltration levels of 10 immune cell types from RNA seq data. Therefore, the score of quanTIseq allows inter- or intra-samples comparison. quanTIseq provides an entire pipeline that can either process the pre-computed gene expression matrix (TPM matrix) or directly analyze bulk RNA-Seq data.  RIMA uses the TPM matrix generated in the proprocessing modules for all immune infiltration algorithms.

```{r,eval=FALSE}
#Run quanTIseq 
res_quant = as.data.frame(deconvolute(df, "quantiseq"))

```

## xCell
<a href="https://xcell.ucsf.edu">xCell</a> performs single-sample gene set enrichment analysis (ssGSEA) on expression data to evaluate the enrichment of immune marker genes. These marker genes were validated using cytometry immunophenotyping and in-silico simulations. Note that results of xCell depend on the number of samples, the power of estimation might be lower if your dataset has limited non-heterogeneous samples.

```{r,eval=FALSE}
#Run xCell
res_xcell = as.data.frame(deconvolute(df, "xcell"))

```


## EPIC
<a href="https://gfellerlab.shinyapps.io/EPIC_1-1/">EPIC</a> uses constrained least square regression to estimate six immune cell types, fibroblasts, and endothelial cells. EPIC collects a unique gene expression reference from circulating and tumor-infiltrating cells. Further, it extended its algorithm to evaluate the uncharacterized cancer cell. The score that comes from the EPIC algorithm is an absolute value that can be compared within or across samples.

```{r,eval=FALSE}
#Run EPIC
res_epic = as.data.frame(deconvolute(df, "epic"))
```

## MCP-counter
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1070-5">MCP-counter</a> uses its transcriptomic markers (TM) curated from 81 public datasets to estimate fibroblasts, endothelial, and eight immune cell types. 

```{r, eval=FALSE}
#Run EPIC
res_mcp = as.data.frame(deconvolute(df, "mcp_counter"))
```


## Comparisons across cell type and across samples
Using results from CIBERSORT abs mode as an example, we provide two other functions to perform inter- or intra sample comparison.  The phenotype used for comparison is the one identified in the config.yaml cohort parameter section.  This section is described more detail in Chapter 4.2.1.

### Comparison across cell type
```{R}

#extract the immune cell score from results of CIBERSORT
res_ciber <- res_ciber[,1:22]
res_ciber

#load plotfunction 
source("plotfunction.R")

#load metasheet for RIMA pipeline
metasheet <- read.csv("metasheet.csv")

#methods: CIBERSORT, quanTIseq, EPIC, MCP, xCell, TIMER
hmap(ta = res_ciber, meta = metasheet, methods = "CIBERSORT", phenotype = "Responder")
```

## Comparison across samples
```{R}
boxfig(ta = res_ciber, meta = meta, methods = "CIBERSORT", phenotype = "Responder")
```

## Video demo
