# Pre-processing of bulk RNA-seq data {#Preprocessing}
In this chapter, we will introduce how we align RNA-seq data, check the data quality, quantify gene expression and handle batch effects across samples.    

## Read Mapping
### STAR
<a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">STAR</a> is one of the most common tools used for bulk RNA-seq data alignment to generate transcriptome BAM or genomic BAM output, which can be downloaded at <a href="https://github.com/alexdobin/STAR">https://github.com/alexdobin/STAR </a>. A tutorial of runing **STAR** is available at <a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">tutorial</a>. \
Once you have installed **STAR**, the first step is to create genome index. In our RIMA pipeline, we download the human genome (hg38) STAR index from <a href="https://gdc.cancer.gov/">Genomic Data Commons (GDC) </a>. 

To run STAR of a paired-end sample:
```{r eval=FALSE}
      STAR --runThreadN 4 --genomeDir /path/to/index
        --outReadsUnmapped None 
        --chimSegmentMin 12  #generate Chimeric and circulate alignment
        --outSAMunmapped Within 
        --outSAMtype BAM SortedByCoordinate
        --readFilesIn sample_1.fastq.gz sample_2.fastq.gz
        --readFilesCommand zcat
        --outFileNamePrefix sample
        --quantMode TranscriptomeSAM GeneCounts # generate outputs in transcript coordinates and mapped read counts per gene
```

Example outputs:
```{r eval=FALSE}
sampleAligned.sortedByCoord.out.bam
sampleAligned.toTranscriptome.out.bam 
sampleReadsPerGene.out.tab
sampleChimeric.out.junction 
sampleLog.final.out
```

Then we use samtools to generate statistics of alignment BAM file:
```{r eval=FALSE}
samtools stats sampleAligned.sortedByCoord.out.bam | grep ^SN | cut -f 2- > sample.sorted.bam.stat.txt
```
Example output of BAM summary:
```{r eval=FALSE}
raw total sequences:	60589702
filtered sequences:	0
sequences:	60589702
is sorted:	1
1st fragments:	30294851
last fragments:	30294851
reads mapped:	59963698
reads mapped and paired:	59963698	# paired-end technology bit set + both mates mapped
reads unmapped:	626004
reads properly paired:	59963698	# proper-pair bit set
reads paired:	60589702	# paired-end technology bit set
reads duplicated:	0	# PCR or optical duplicate bit set
reads MQ0:	110872	# mapped and MQ=0
reads QC failed:	0
non-primary alignments:	3102614
total length:	6058970200	# ignores clipping
total first fragment length:	3029485100	# ignores clipping
total last fragment length:	3029485100	# ignores clipping
bases mapped:	5996369800	# ignores clipping
bases mapped (cigar):	5952761135	# more accurate
bases trimmed:	0
bases duplicated:	0
mismatches:	0	# from NM fields
error rate:	0.000000e+00	# mismatches / bases mapped (cigar)
average length:	100
average first fragment length:	100
average last fragment length:	100
maximum length:	100
maximum first fragment length:	100
maximum last fragment length:	100
average quality:	39.7
insert size average:	971.3
insert size standard deviation:	1899.5
inward oriented pairs:	29281286
outward oriented pairs:	700563
pairs with other orientation:	0
pairs on different chromosomes:	0
```
## Quality Control
RseQC is a standard tool for checking the quality of reads alignment, providing principal measurements of RNA-seq data quality. RseQC tutorial is available at <a href="http://rseqc.sourceforge.net/">here</a>. \

To reduce running time, we first subsampled the BAM file and use only housekeeping genes for quality check using RseQC.
```{r eval=FALSE}
## Example of subsampling 50% of input alignment
samtools view -s 0.5 -b sampleAligned.sortedByCoord.out.bam  > sample_downsampling.bam 
# Extract the alignment of housekeeping gene.
bedtools intersect -a sample_downsampling.bam -b housekeeping_refseqGenes.bed > sample_downsampling_housekeeping.bam 
# index BAM
samtools index sample_downsampling_housekeeping.bam > sample_downsampling_housekeeping.bam.bai 
```
### Quality metrics

1. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0922-z#Sec2">Transcript integrity number (TIN)</a> is the most widely used measure RNA integrity, which is the percentage of transcripts that has uniform read coverage across the genome. RseQC calculated the TIN of each transcripts and the <a href="http://rseqc.sourceforge.net/#tin-py">median TIN score (medTIN) </a> across all transcripts has is commonly used to indicate the RNA integrity and of each sample. \
2. <a href="http://rseqc.sourceforge.net/#read-distribution-py">Read distribution </a> summarizes the fraction of reads aligned into different genomic regions, such as exon and intron regions. \
3. <a href="http://rseqc.sourceforge.net/#genebody-coverage-py">Gene body coverage</a> shows the RNA-seq read coverage over the gene body. \
4. <a href="http://rseqc.sourceforge.net/#junction-saturation-py">Junction saturation</a> detects the splicing junctions with different resampling percentages of reads to determine if the sequencing depth is sufficient to perform alternative splicing analysis. \

Samples with low alignment fraction or low integrity or abnormal reads distribution usually have bad quality and should be removed in downstream analysis. \

## Gene quantification
After the alignment of sequencing reads and quality check, we will quantify the gene or transcript expression from the BAM file.Both <a href="https://combine-lab.github.io/salmon/getting_started/">Salmon</a> and <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323">RSEM</a> are widely used for gene quantification. Salmon conducts fast transcript-level quantification, and RSEM performed both gene-level and transcript-level quantification. Salmon is usually faster and less memory-consuming than RSEM. Both tools generate transcripts per million (TPM), reads per kilobase of exon model per million reads (RPKM) and fragments per kilobase of exon model per million mapped reads (FPKM) expression, which is the normalized gene quantification that removing the effect of different gene length. In addition to TPM, <a href="https://htseq.readthedocs.io/en/master/tour.html#counting-reads-by-genes">HTSeq </a> quantifies gene expression according to mapped read counts. Both normalized gene expression and read counts can be used for differential expression analysis. \

## Batch effect removal
Batch effects across samples is easily ignored but worth considering problems for immunotherapy cohort study. It is usually caused by unbalanced experimental design where estimation of group difference, and they are interdependent. To avoid the actual biology variation confounded by batch effect, <a href="https://www.degruyter.com/document/doi/10.2202/1544-6115.1027/html"> Limma </a> and <a href="https://academic.oup.com/biostatistics/article/8/1/118/252073?login=true">Combat</a> are common approaches to correct the batch effects. Limma uses a two-way ANOVA approach, while Combat uses the empirical Bayes approach, which is critical for small batches to avoid over-correction. For large batches, both methods should be similar. The <a href="https://bioconductor.org/packages/release/bioc/html/sva.html">sva</a> R package integrates both methods for batch effect correction.
To check whether the batch effect has been removed,  principal components analysis (PCA) or unsupervised clustering before and after batch effect removal will be an excellent way to validate. To ensure a solid downstream RNA-seq analysis, it is essential to generate a gene expression profile mitigating the batch effect and group difference from a well-designed study.

Example of PCA before and after batch effect between FT and FFPE samples:
```{r fig.align='center', echo=FALSE, fig.cap='Before(left) and after(right) batch correction with Limma'}
knitr::include_graphics('images/PCA_batch.png', dpi = NA)
```

## Video demo of RIMA

![RIMA preprocess module demo](videos/preprocess.mp4)
