<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Immune Infiltration | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Immune Infiltration | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Immune Infiltration | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  



<meta name="date" content="2021-04-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Repertoire.html"/>
<link rel="next" href="Response.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this tutorial</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#pre-built-reference-data"><i class="fa fa-check"></i><b>1.2</b> Pre-built reference data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-mapping"><i class="fa fa-check"></i><b>3.1</b> Read Mapping</a><ul>
<li class="chapter" data-level="3.1.1" data-path="Preprocessing.html"><a href="Preprocessing.html#star"><i class="fa fa-check"></i><b>3.1.1</b> STAR</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a><ul>
<li class="chapter" data-level="3.2.1" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-metrics"><i class="fa fa-check"></i><b>3.2.1</b> Quality metrics</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a></li>
<li class="chapter" data-level="5" data-path="Immune.html"><a href="Immune.html"><i class="fa fa-check"></i><b>5</b> Immune Analysis</a></li>
<li class="chapter" data-level="6" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>6</b> Immune repertoire</a></li>
<li class="chapter" data-level="7" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>7</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="7.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>7.1</b> Cibersort</a></li>
<li class="chapter" data-level="7.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>7.2</b> TIMER</a></li>
<li class="chapter" data-level="7.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>7.3</b> quanTIseq</a></li>
<li class="chapter" data-level="7.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>7.4</b> xCell</a></li>
<li class="chapter" data-level="7.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>7.5</b> EPIC</a></li>
<li class="chapter" data-level="7.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>7.6</b> MCP-counter</a></li>
<li class="chapter" data-level="7.7" data-path="Infiltration.html"><a href="Infiltration.html#video-demo"><i class="fa fa-check"></i><b>7.7</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>8</b> Immune Response</a></li>
<li class="chapter" data-level="9" data-path="Neoantigen.html"><a href="Neoantigen.html"><i class="fa fa-check"></i><b>9</b> Neoantigen Analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="Neoantigen.html"><a href="Neoantigen.html#hla-calling"><i class="fa fa-check"></i><b>9.1</b> HLA calling</a></li>
<li class="chapter" data-level="9.2" data-path="Neoantigen.html"><a href="Neoantigen.html#somatic-mutation"><i class="fa fa-check"></i><b>9.2</b> Somatic mutation</a></li>
<li class="chapter" data-level="9.3" data-path="Neoantigen.html"><a href="Neoantigen.html#neoantigen-identification"><i class="fa fa-check"></i><b>9.3</b> Neoantigen identification</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>10</b> Fusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Infiltration" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Immune Infiltration</h1>
<p><strong>Cell infiltration estimation in the tumor microenvironment</strong><br />
TME is a complicated system consisting of tumor cells, immune cells, stromal cells, and extracellular factors. Estimating cell components is essential for classifying the distinct tumor immune microenvironment (TIME) phenotype. Furthermore, dissecting the TIME by evaluating cell components plays a significant role in untangling the mechanism of tumor progression and immune evasion, which is essential for providing a guideline for immunotherapy selection. There are two major approaches for estimating cell infiltration in TME, deconvolution-based and marker-based approaches. The deconvolution-based method infers cell type fractions from the bulk gene expression of a given list of signature genes, such as TIMER, quanTIseq, EPIC, and CIBERSORT. The marker-based approaches to quantify the enrichment score utilizing a list of cell-specific marker genes, such as xCell and MCP-counter. <a href="https://github.com/icbi-lab/immunedeconv">ImmuneDeconv</a> is an R package that integrated these six methods. All six algorithms estimate cell infiltration of immune cells and stromal cells, while EPIC and quanTIseq make it available to assess uncharacterized cells defined as cancer cells. Of note, CIBERSORT absolute mode, EPIC, and quanTIseq support inter-comparison between sample groups and intra-comparison between cell types. And TIMER, xCell, and MCP-counter support only inter-comparison between sample groups for one cell type. Immune cell deconvolution is still an open question in the field, and different algorithms could give slightly different results. Each algorithm estimates one general cell type or specific cell subtype. The best way is to focus on the observations consistently predicted by various algorithms, in which case the gold standard is not available.<br />
</p>
<div id="cibersort" class="section level2">
<h2><span class="header-section-number">7.1</span> Cibersort</h2>
<p><a href="https://www.nature.com/articles/nmeth.3337">CIBERSORT</a> is a deconvolution-based method that uses the v-support vector regression method (v-SVR) to estimate each immune cell type’s relative proportion from gene expression profile (GEP). It requires an input of reference gene expression signatures. CIBERSORT provided a gene signature matrix (LM22.txt) which contains 547 genes distinguish 7 T-cell types, naïve and memory B cells, plasma cell, and NK cells. The CIBERSORT absolute mode scales cellular fraction to a score that reflects each cell type’s absolute proportion, allowing comparison across both samples and cell types.
The <strong>CIBERSORT source code</strong> and <strong>LM22 signature matrix</strong> are only freely available to academic users, which can be applied through <a href="https://cibersort.stanford.edu/">CIBERSORT website </a>.<br />
</p>
<p>To run CIBERSORT using ImmuneDeconv:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="Infiltration.html#cb6-1"></a><span class="co">#demo</span></span>
<span id="cb6-2"><a href="Infiltration.html#cb6-2"></a><span class="kw">suppressMessages</span>(<span class="kw">library</span>(immunedeconv))</span>
<span id="cb6-3"><a href="Infiltration.html#cb6-3"></a><span class="co">#read gene expression matrix </span></span>
<span id="cb6-4"><a href="Infiltration.html#cb6-4"></a>input &lt;-<span class="st"> &quot;tpm_convertID.txt&quot;</span></span>
<span id="cb6-5"><a href="Infiltration.html#cb6-5"></a></span>
<span id="cb6-6"><a href="Infiltration.html#cb6-6"></a><span class="co">#load CIBERSORT source code and signiture matrix </span></span>
<span id="cb6-7"><a href="Infiltration.html#cb6-7"></a><span class="kw">source</span>(<span class="st">&quot;CIBERSORT.R&quot;</span>)</span>
<span id="cb6-8"><a href="Infiltration.html#cb6-8"></a>sig_matrix &lt;-<span class="st"> &quot;LM22.txt&quot;</span></span>
<span id="cb6-9"><a href="Infiltration.html#cb6-9"></a></span>
<span id="cb6-10"><a href="Infiltration.html#cb6-10"></a><span class="co">#Run CIBERSORT abs </span></span>
<span id="cb6-11"><a href="Infiltration.html#cb6-11"></a><span class="co">#The number of permutation</span></span>
<span id="cb6-12"><a href="Infiltration.html#cb6-12"></a>cibersort_perm =<span class="st"> </span><span class="dv">100</span></span>
<span id="cb6-13"><a href="Infiltration.html#cb6-13"></a><span class="co">#Quantile normalization of input mixture, default = FALSE for RNA-Seq data</span></span>
<span id="cb6-14"><a href="Infiltration.html#cb6-14"></a>cibersort_qn =<span class="st"> </span><span class="ot">FALSE</span></span>
<span id="cb6-15"><a href="Infiltration.html#cb6-15"></a><span class="co">#whether to apply absolute mode in cibersort</span></span>
<span id="cb6-16"><a href="Infiltration.html#cb6-16"></a>cibersort_abs =<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb6-17"><a href="Infiltration.html#cb6-17"></a><span class="co">#sig.score = for each mixture sample, define S as the median expression,level of all genes in the signature matrix divided by the median expression level of all genes in the mixture. Multiple cell subset fractions by S.</span></span>
<span id="cb6-18"><a href="Infiltration.html#cb6-18"></a>cibersort_abs_method =<span class="st"> &quot;sig.score&quot;</span></span>
<span id="cb6-19"><a href="Infiltration.html#cb6-19"></a>res_ciber &lt;-<span class="st"> </span><span class="kw">CIBERSORT</span>(sig_matrix, input, <span class="dt">perm =</span> cibersort_perm, <span class="dt">QN =</span> cibersort_qn, <span class="dt">absolute =</span> cibersort_abs,</span>
<span id="cb6-20"><a href="Infiltration.html#cb6-20"></a>                       <span class="dt">abs_method =</span> cibersort_abs_method)</span>
<span id="cb6-21"><a href="Infiltration.html#cb6-21"></a></span>
<span id="cb6-22"><a href="Infiltration.html#cb6-22"></a><span class="kw">head</span>(res_ciber,<span class="dv">3</span>)</span></code></pre></div>
<pre><code>##            B cells naive B cells memory Plasma cells T cells CD8
## SRR8281218   0.002023271    0.003060637  0.000000000  0.12297118
## SRR8281219   0.010693087    0.000000000  0.001201108  0.38499197
## SRR8281226   0.000000000    0.008281786  0.000000000  0.01442307
##            T cells CD4 naive T cells CD4 memory resting
## SRR8281218                 0                 0.14975225
## SRR8281219                 0                 0.22216858
## SRR8281226                 0                 0.08219228
##            T cells CD4 memory activated T cells follicular helper
## SRR8281218                            0                0.08924869
## SRR8281219                            0                0.16924345
## SRR8281226                            0                0.06671124
##            T cells regulatory (Tregs) T cells gamma delta NK cells resting
## SRR8281218                 0.09617560                   0       0.06763887
## SRR8281219                 0.00000000                   0       0.00000000
## SRR8281226                 0.04296535                   0       0.07211706
##            NK cells activated  Monocytes Macrophages M0 Macrophages M1
## SRR8281218          0.0000000 0.06195550      0.0000000     0.01550602
## SRR8281219          0.1076938 0.03659288      0.1552708     0.03207925
## SRR8281226          0.0357515 0.41486727      0.0000000     0.00000000
##            Macrophages M2 Dendritic cells resting Dendritic cells activated
## SRR8281218      0.7049077                       0               0.000000000
## SRR8281219      1.1589577                       0               0.000000000
## SRR8281226      0.6474766                       0               0.004669831
##            Mast cells resting Mast cells activated Eosinophils Neutrophils
## SRR8281218         0.11994310                    0 0.003766691  0.00000000
## SRR8281219         0.24218289                    0 0.000000000  0.07942792
## SRR8281226         0.01289667                    0 0.002926060  0.12361329
##            P-value Correlation     RMSE Absolute score (sig.score)
## SRR8281218    0.04  0.15699314 1.027475                   1.436950
## SRR8281219    0.24  0.05195861 1.072383                   2.600503
## SRR8281226    0.05  0.13503937 1.054532                   1.528892</code></pre>
</div>
<div id="timer" class="section level2">
<h2><span class="header-section-number">7.2</span> TIMER</h2>
<p><a href="http://cistrome.org/TIMER/">TIMER</a> uses linear least square regression to estimate six tumor-infiltrating immune cell types (B, CD4T, CD8T, neutrophils, macrophages, and dendritic cells). Since the tumor purity is very different across different cancer cell types, one of key steps is that TIMER selected the specific genes as immune signatures based on the correlation between gene expression and tumor purity. Note that the TIMER score can only be compared within samples, and the current version of TIMER supports 23 different TCGA cancer types.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="Infiltration.html#cb8-1"></a><span class="co">#load gene expression matrix </span></span>
<span id="cb8-2"><a href="Infiltration.html#cb8-2"></a>df &lt;-<span class="st"> </span><span class="kw">read.table</span>(input, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="dt">row.names =</span> <span class="dv">1</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb8-3"><a href="Infiltration.html#cb8-3"></a></span>
<span id="cb8-4"><a href="Infiltration.html#cb8-4"></a><span class="co">#The TCGA cancer acronyms of demo data</span></span>
<span id="cb8-5"><a href="Infiltration.html#cb8-5"></a>cancertype &lt;-<span class="st"> &quot;GBM&quot;</span></span>
<span id="cb8-6"><a href="Infiltration.html#cb8-6"></a><span class="co">#Available cancer type </span></span>
<span id="cb8-7"><a href="Infiltration.html#cb8-7"></a>immunedeconv<span class="op">::</span>timer_available_cancers</span>
<span id="cb8-8"><a href="Infiltration.html#cb8-8"></a>res_timer =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;timer&quot;</span>,<span class="dt">indications=</span><span class="kw">rep</span>(<span class="kw">tolower</span>(cancertype),<span class="kw">ncol</span>(df))))</span></code></pre></div>
</div>
<div id="quantiseq" class="section level2">
<h2><span class="header-section-number">7.3</span> quanTIseq</h2>
<p>The <a href="https://icbi.i-med.ac.at/software/quantiseq/doc/index.html">quanTIseq</a> uses constrained least square regression to estimate absolute proportions of 10 immune cell types infiltration levels from RNA seq data. Therefore, the score of quanTIseq allows inter- or intra-samples comparison. quanTIseq provides an entire pipeline that can either process the pre-computed gene expression matrix (TPM matrix) or directly analyze bulk RNA-Seq data.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="Infiltration.html#cb9-1"></a><span class="co">#Run quanTIseq </span></span>
<span id="cb9-2"><a href="Infiltration.html#cb9-2"></a>res_quant =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;quantiseq&quot;</span>))</span></code></pre></div>
</div>
<div id="xcell" class="section level2">
<h2><span class="header-section-number">7.4</span> xCell</h2>
<p><a href="https://xcell.ucsf.edu">xCell</a> performs single-sample gene set enrichment analysis (ssGSEA) on expression data to evaluate the enrichment of immune marker genes. These marker genes validated using cytometry immunophenotyping and in-silico simulations. Notes that results of xCell depend on the number of samples, the power of estimation might be lower if providing limited non-heterogeneous samples</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="Infiltration.html#cb10-1"></a><span class="co">#Run xCell</span></span>
<span id="cb10-2"><a href="Infiltration.html#cb10-2"></a>res_xcell =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;xcell&quot;</span>))</span></code></pre></div>
</div>
<div id="epic" class="section level2">
<h2><span class="header-section-number">7.5</span> EPIC</h2>
<p><a href="https://gfellerlab.shinyapps.io/EPIC_1-1/">EPIC</a> uses constrained least square regression to estimate six immune cell types, fibroblasts, and endothelial cells. EPIC collects a unique gene expression reference from circulating and tumor-infiltrating cells. Further, it extended its algorithm to evaluate the uncharacterized cancer cell. The score that comes from the EPIC algorithm is the absolute value that can be compared within or across samples.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="Infiltration.html#cb11-1"></a><span class="co">#Run EPIC</span></span>
<span id="cb11-2"><a href="Infiltration.html#cb11-2"></a>res_epic =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;epic&quot;</span>))</span></code></pre></div>
</div>
<div id="mcp-counter" class="section level2">
<h2><span class="header-section-number">7.6</span> MCP-counter</h2>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1070-5">MCP-counter</a> uses its transcriptomic markers (TM) curated from 81 public datasets to estimate fibroblasts, endothelial, and eight immune cell types. The difference of mean log2-expression of marker genes samples is a key to evaluate the immune cell infiltration levels.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="Infiltration.html#cb12-1"></a><span class="co">#Run EPIC</span></span>
<span id="cb12-2"><a href="Infiltration.html#cb12-2"></a>res_mcp =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;mcp_counter&quot;</span>))</span></code></pre></div>
</div>
<div id="video-demo" class="section level2">
<h2><span class="header-section-number">7.7</span> Video demo</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Repertoire.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Response.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
