<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Pre-processing of bulk RNA-seq data | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Pre-processing of bulk RNA-seq data | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Pre-processing of bulk RNA-seq data | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  



<meta name="date" content="2021-04-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="Differential.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this tutorial</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#pre-built-reference-data"><i class="fa fa-check"></i><b>1.2</b> Pre-built reference data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-mapping"><i class="fa fa-check"></i><b>3.1</b> Read Mapping</a><ul>
<li class="chapter" data-level="3.1.1" data-path="Preprocessing.html"><a href="Preprocessing.html#star"><i class="fa fa-check"></i><b>3.1.1</b> STAR</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a><ul>
<li class="chapter" data-level="3.2.1" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-metrics"><i class="fa fa-check"></i><b>3.2.1</b> Quality metrics</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-cell-type"><i class="fa fa-check"></i><b>6.7</b> Comparison across cell type</a></li>
<li class="chapter" data-level="6.8" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-sample"><i class="fa fa-check"></i><b>6.8</b> Comparison across sample</a></li>
<li class="chapter" data-level="6.9" data-path="Infiltration.html"><a href="Infiltration.html#video-demo"><i class="fa fa-check"></i><b>6.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.1</b> TIDE</a><ul>
<li class="chapter" data-level="7.1.1" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.1.1</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.1.2" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.1.2</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.1.3" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.1.3</b> Microsatelite instability (MSI)</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#comparison-between-responders-and-non-reponders"><i class="fa fa-check"></i><b>7.2</b> Comparison between responders and non-reponders</a></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#video-demo-1"><i class="fa fa-check"></i><b>7.3</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Neoantigen.html"><a href="Neoantigen.html"><i class="fa fa-check"></i><b>8</b> Neoantigen Analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="Neoantigen.html"><a href="Neoantigen.html#hla-calling"><i class="fa fa-check"></i><b>8.1</b> HLA calling</a></li>
<li class="chapter" data-level="8.2" data-path="Neoantigen.html"><a href="Neoantigen.html#somatic-mutation"><i class="fa fa-check"></i><b>8.2</b> Somatic mutation</a></li>
<li class="chapter" data-level="8.3" data-path="Neoantigen.html"><a href="Neoantigen.html#neoantigen-identification"><i class="fa fa-check"></i><b>8.3</b> Neoantigen identification</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>9</b> Fusion</a><ul>
<li class="chapter" data-level="9.1" data-path="fusion.html"><a href="fusion.html#fusion-calling-from-chimeric"><i class="fa fa-check"></i><b>9.1</b> Fusion calling from CHIMERIC</a></li>
<li class="chapter" data-level="9.2" data-path="fusion.html"><a href="fusion.html#remove-homolog"><i class="fa fa-check"></i><b>9.2</b> Remove homolog</a></li>
<li class="chapter" data-level="9.3" data-path="fusion.html"><a href="fusion.html#comparison"><i class="fa fa-check"></i><b>9.3</b> Comparison</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Preprocessing" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Pre-processing of bulk RNA-seq data</h1>
<p>In this chapter, we will introduce how we align RNA-seq data, check the data quality, quantify gene expression and handle batch effects across samples.</p>
<div id="read-mapping" class="section level2">
<h2><span class="header-section-number">3.1</span> Read Mapping</h2>
<div id="star" class="section level3">
<h3><span class="header-section-number">3.1.1</span> STAR</h3>
<p><a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">STAR</a> is one of the most common tools used for bulk RNA-seq data alignment to generate transcriptome BAM or genomic BAM output, which can be downloaded at <a href="https://github.com/alexdobin/STAR"> here</a>. A tutorial of runing STAR is available at <a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">here</a>. Once you have installed STAR, the first step is to create genome index. In our RIMA pipeline, we download the human genome (hg38) STAR index from <a href="https://gdc.cancer.gov/">Genomic Data Commons (GDC) </a>.</p>
<p>To run STAR of a sample:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="Preprocessing.html#cb1-1"></a>      STAR <span class="op">--</span>runThreadN <span class="dv">4</span> <span class="op">--</span>genomeDir <span class="op">/</span>path<span class="op">/</span>to<span class="op">/</span>index</span>
<span id="cb1-2"><a href="Preprocessing.html#cb1-2"></a>        <span class="op">--</span>outReadsUnmapped None </span>
<span id="cb1-3"><a href="Preprocessing.html#cb1-3"></a>        <span class="op">--</span>chimSegmentMin <span class="dv">12</span>  <span class="co">#generate Chimeric and circulate alignment</span></span>
<span id="cb1-4"><a href="Preprocessing.html#cb1-4"></a>        <span class="op">--</span>outSAMunmapped Within </span>
<span id="cb1-5"><a href="Preprocessing.html#cb1-5"></a>        <span class="op">--</span>outSAMtype BAM SortedByCoordinate</span>
<span id="cb1-6"><a href="Preprocessing.html#cb1-6"></a>        <span class="op">--</span>readFilesIn sample_<span class="fl">1.</span>fastq.gz sample_<span class="fl">2.</span>fastq.gz</span>
<span id="cb1-7"><a href="Preprocessing.html#cb1-7"></a>        <span class="op">--</span>readFilesCommand zcat</span>
<span id="cb1-8"><a href="Preprocessing.html#cb1-8"></a>        <span class="op">--</span>outFileNamePrefix sample</span>
<span id="cb1-9"><a href="Preprocessing.html#cb1-9"></a>        <span class="op">--</span>quantMode TranscriptomeSAM GeneCounts <span class="co"># generate outputs in transcript coordinates and mapped read counts per gene</span></span></code></pre></div>
<p>Example outputs:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="Preprocessing.html#cb2-1"></a>sampleAligned.sortedByCoord.out.bam</span>
<span id="cb2-2"><a href="Preprocessing.html#cb2-2"></a>sampleAligned.toTranscriptome.out.bam </span>
<span id="cb2-3"><a href="Preprocessing.html#cb2-3"></a>sampleReadsPerGene.out.tab</span>
<span id="cb2-4"><a href="Preprocessing.html#cb2-4"></a>sampleChimeric.out.junction </span>
<span id="cb2-5"><a href="Preprocessing.html#cb2-5"></a>sampleLog.final.out</span></code></pre></div>
<p>Then we use samtools to generate statistics of alignment BAM file:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="Preprocessing.html#cb3-1"></a>samtools stats sampleAligned.sortedByCoord.out.bam <span class="op">|</span><span class="st"> </span>grep <span class="op">^</span>SN <span class="op">|</span><span class="st"> </span>cut <span class="op">-</span>f <span class="dv">2</span><span class="op">-</span><span class="st"> </span><span class="er">&gt;</span><span class="st"> </span>sample.sorted.bam.stat.txt</span></code></pre></div>
<p>Example output of BAM summary:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="Preprocessing.html#cb4-1"></a>raw total sequences<span class="op">:</span><span class="st">    </span><span class="dv">60589702</span></span>
<span id="cb4-2"><a href="Preprocessing.html#cb4-2"></a>filtered sequences<span class="op">:</span><span class="st"> </span><span class="dv">0</span></span>
<span id="cb4-3"><a href="Preprocessing.html#cb4-3"></a>sequences<span class="op">:</span><span class="st">  </span><span class="dv">60589702</span></span>
<span id="cb4-4"><a href="Preprocessing.html#cb4-4"></a>is sorted<span class="op">:</span><span class="st">  </span><span class="dv">1</span></span>
<span id="cb4-5"><a href="Preprocessing.html#cb4-5"></a>1st fragments<span class="op">:</span><span class="st">  </span><span class="dv">30294851</span></span>
<span id="cb4-6"><a href="Preprocessing.html#cb4-6"></a>last fragments<span class="op">:</span><span class="st"> </span><span class="dv">30294851</span></span>
<span id="cb4-7"><a href="Preprocessing.html#cb4-7"></a>reads mapped<span class="op">:</span><span class="st">   </span><span class="dv">59963698</span></span>
<span id="cb4-8"><a href="Preprocessing.html#cb4-8"></a>reads mapped and paired<span class="op">:</span><span class="st">    </span><span class="dv">59963698</span>    <span class="co"># paired-end technology bit set + both mates mapped</span></span>
<span id="cb4-9"><a href="Preprocessing.html#cb4-9"></a>reads unmapped<span class="op">:</span><span class="st"> </span><span class="dv">626004</span></span>
<span id="cb4-10"><a href="Preprocessing.html#cb4-10"></a>reads properly paired<span class="op">:</span><span class="st">  </span><span class="dv">59963698</span>    <span class="co"># proper-pair bit set</span></span>
<span id="cb4-11"><a href="Preprocessing.html#cb4-11"></a>reads paired<span class="op">:</span><span class="st">   </span><span class="dv">60589702</span>    <span class="co"># paired-end technology bit set</span></span>
<span id="cb4-12"><a href="Preprocessing.html#cb4-12"></a>reads duplicated<span class="op">:</span><span class="st">   </span><span class="dv">0</span>   <span class="co"># PCR or optical duplicate bit set</span></span>
<span id="cb4-13"><a href="Preprocessing.html#cb4-13"></a>reads MQ0<span class="op">:</span><span class="st">  </span><span class="dv">110872</span>  <span class="co"># mapped and MQ=0</span></span>
<span id="cb4-14"><a href="Preprocessing.html#cb4-14"></a>reads QC failed<span class="op">:</span><span class="st">    </span><span class="dv">0</span></span>
<span id="cb4-15"><a href="Preprocessing.html#cb4-15"></a>non<span class="op">-</span>primary alignments<span class="op">:</span><span class="st"> </span><span class="dv">3102614</span></span>
<span id="cb4-16"><a href="Preprocessing.html#cb4-16"></a>total length<span class="op">:</span><span class="st">   </span><span class="dv">6058970200</span>  <span class="co"># ignores clipping</span></span>
<span id="cb4-17"><a href="Preprocessing.html#cb4-17"></a>total first fragment length<span class="op">:</span><span class="st">    </span><span class="dv">3029485100</span>  <span class="co"># ignores clipping</span></span>
<span id="cb4-18"><a href="Preprocessing.html#cb4-18"></a>total last fragment length<span class="op">:</span><span class="st"> </span><span class="dv">3029485100</span>  <span class="co"># ignores clipping</span></span>
<span id="cb4-19"><a href="Preprocessing.html#cb4-19"></a>bases mapped<span class="op">:</span><span class="st">   </span><span class="dv">5996369800</span>  <span class="co"># ignores clipping</span></span>
<span id="cb4-20"><a href="Preprocessing.html#cb4-20"></a>bases <span class="kw">mapped</span> (cigar)<span class="op">:</span><span class="st">   </span><span class="dv">5952761135</span>  <span class="co"># more accurate</span></span>
<span id="cb4-21"><a href="Preprocessing.html#cb4-21"></a>bases trimmed<span class="op">:</span><span class="st">  </span><span class="dv">0</span></span>
<span id="cb4-22"><a href="Preprocessing.html#cb4-22"></a>bases duplicated<span class="op">:</span><span class="st">   </span><span class="dv">0</span></span>
<span id="cb4-23"><a href="Preprocessing.html#cb4-23"></a>mismatches<span class="op">:</span><span class="st"> </span><span class="dv">0</span>   <span class="co"># from NM fields</span></span>
<span id="cb4-24"><a href="Preprocessing.html#cb4-24"></a>error rate<span class="op">:</span><span class="st"> </span><span class="fl">0.000000e+00</span>    <span class="co"># mismatches / bases mapped (cigar)</span></span>
<span id="cb4-25"><a href="Preprocessing.html#cb4-25"></a>average length<span class="op">:</span><span class="st"> </span><span class="dv">100</span></span>
<span id="cb4-26"><a href="Preprocessing.html#cb4-26"></a>average first fragment length<span class="op">:</span><span class="st">  </span><span class="dv">100</span></span>
<span id="cb4-27"><a href="Preprocessing.html#cb4-27"></a>average last fragment length<span class="op">:</span><span class="st">   </span><span class="dv">100</span></span>
<span id="cb4-28"><a href="Preprocessing.html#cb4-28"></a>maximum length<span class="op">:</span><span class="st"> </span><span class="dv">100</span></span>
<span id="cb4-29"><a href="Preprocessing.html#cb4-29"></a>maximum first fragment length<span class="op">:</span><span class="st">  </span><span class="dv">100</span></span>
<span id="cb4-30"><a href="Preprocessing.html#cb4-30"></a>maximum last fragment length<span class="op">:</span><span class="st">   </span><span class="dv">100</span></span>
<span id="cb4-31"><a href="Preprocessing.html#cb4-31"></a>average quality<span class="op">:</span><span class="st">    </span><span class="fl">39.7</span></span>
<span id="cb4-32"><a href="Preprocessing.html#cb4-32"></a>insert size average<span class="op">:</span><span class="st">    </span><span class="fl">971.3</span></span>
<span id="cb4-33"><a href="Preprocessing.html#cb4-33"></a>insert size standard deviation<span class="op">:</span><span class="st"> </span><span class="fl">1899.5</span></span>
<span id="cb4-34"><a href="Preprocessing.html#cb4-34"></a>inward oriented pairs<span class="op">:</span><span class="st">  </span><span class="dv">29281286</span></span>
<span id="cb4-35"><a href="Preprocessing.html#cb4-35"></a>outward oriented pairs<span class="op">:</span><span class="st"> </span><span class="dv">700563</span></span>
<span id="cb4-36"><a href="Preprocessing.html#cb4-36"></a>pairs with other orientation<span class="op">:</span><span class="st">   </span><span class="dv">0</span></span>
<span id="cb4-37"><a href="Preprocessing.html#cb4-37"></a>pairs on different chromosomes<span class="op">:</span><span class="st"> </span><span class="dv">0</span></span></code></pre></div>
</div>
</div>
<div id="quality-control" class="section level2">
<h2><span class="header-section-number">3.2</span> Quality Control</h2>
<p>RseQC is a standard tool for checking the quality of reads alignment, providing principal measurements of RNA-seq data quality. RseQC tutorial is available at <a href="http://rseqc.sourceforge.net/">here</a>.<br />
</p>
<p>To reduce running time, we first subsampled the BAM file and use only housekeeping genes for quality check using RseQC.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="Preprocessing.html#cb5-1"></a><span class="co">## Example of subsampling 50% of input alignment</span></span>
<span id="cb5-2"><a href="Preprocessing.html#cb5-2"></a>samtools view <span class="op">-</span>s <span class="fl">0.5</span> <span class="op">-</span>b sampleAligned.sortedByCoord.out.bam  <span class="op">&gt;</span><span class="st"> </span>sample_downsampling.bam </span>
<span id="cb5-3"><a href="Preprocessing.html#cb5-3"></a><span class="co"># Extract the alignment of housekeeping gene.</span></span>
<span id="cb5-4"><a href="Preprocessing.html#cb5-4"></a>bedtools intersect <span class="op">-</span>a sample_downsampling.bam <span class="op">-</span>b housekeeping_refseqGenes.bed <span class="op">&gt;</span><span class="st"> </span>sample_downsampling_housekeeping.bam </span>
<span id="cb5-5"><a href="Preprocessing.html#cb5-5"></a><span class="co"># index BAM</span></span>
<span id="cb5-6"><a href="Preprocessing.html#cb5-6"></a>samtools index sample_downsampling_housekeeping.bam <span class="op">&gt;</span><span class="st"> </span>sample_downsampling_housekeeping.bam.bai </span></code></pre></div>
<div id="quality-metrics" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Quality metrics</h3>
<ol style="list-style-type: decimal">
<li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0922-z#Sec2">Transcript integrity number (TIN)</a> is the most widely used measure RNA integrity, which is the percentage of transcripts that has uniform read coverage across the genome. RseQC calculated the TIN each transcripts and the <a href="http://rseqc.sourceforge.net/#tin-py">median TIN score (medTIN) </a> across all transcripts has is commonly used to indicate the RNA integrity and of each sample.<br />
</li>
<li><a href="http://rseqc.sourceforge.net/#read-distribution-py">Read distribution </a> summarizes the fraction of reads aligned into different genomic regions, such as exon and intron regions.<br />
</li>
<li><a href="http://rseqc.sourceforge.net/#genebody-coverage-py">Gene body coverage</a> shows the RNA-seq read coverage over the gene body.<br />
</li>
<li><a href="http://rseqc.sourceforge.net/#junction-saturation-py">Junction saturation</a> detects the splicing junctions with different resampling percentages of reads to determine if the sequencing depth is sufficient to perform alternative splicing analysis.<br />
</li>
</ol>
<p>Samples with low alignment fraction or low integrity or abnormal reads distribution usually have bad quality and should be removed in downstream analysis.<br />
</p>
</div>
</div>
<div id="gene-quantification" class="section level2">
<h2><span class="header-section-number">3.3</span> Gene quantification</h2>
<p>After the alignment of sequencing reads and quality check, we will quantify the gene or transcript expression from the BAM file.Both <a href="https://combine-lab.github.io/salmon/getting_started/">Salmon</a> and <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323">RSEM</a> are widely used for gene quantification. Salmon conducts fast transcript-level quantification, and RSEM performed both gene-level and transcript-level quantification. Salmon is usually faster and less memory-consuming than RSEM. Both tools generate transcripts per million (TPM), reads per kilobase of exon model per million reads (RPKM) and fragments per kilobase of exon model per million mapped reads (FPKM) expression, which is the normalized gene quantification that removing the effect of different gene length. In addition to TPM, <a href="https://htseq.readthedocs.io/en/master/tour.html#counting-reads-by-genes">HTSeq </a> and STAR quantify gene expression according to mapped read counts. Both normalized gene expression and read counts can be used for differential expression analysis.<br />
</p>
</div>
<div id="batch-effect-removal" class="section level2">
<h2><span class="header-section-number">3.4</span> Batch effect removal</h2>
<p>Batch effects across samples is easily ignored but worth considering problems for immunotherapy cohort study. It is usually caused by unbalanced experimental design where estimation of group difference, and they are interdependent. To avoid the actual biology variation confounded by batch effect, <a href="https://www.degruyter.com/document/doi/10.2202/1544-6115.1027/html"> Limma </a> and <a href="https://academic.oup.com/biostatistics/article/8/1/118/252073?login=true">Combat</a> are common approaches to correct the batch effects. Limma uses a two-way ANOVA approach, while Combat uses the empirical Bayes approach, which is critical for small batches to avoid over-correction. For large batches, both methods should be similar. The <a href="https://bioconductor.org/packages/release/bioc/html/sva.html">sva</a> R package integrates both methods for batch effect correction.
To check whether the batch effect has been removed, principal components analysis (PCA) or unsupervised clustering before and after batch effect removal will be an excellent way to validate. To ensure a solid downstream RNA-seq analysis, it is essential to generate a gene expression profile mitigating the batch effect and group difference from a well-designed study.</p>
Example of PCA before and after batch effect between FT and FFPE samples:
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-8"></span>
<img src="images/PCA_batch.png" alt="Before(left) and after(right) batch correction with Limma"  />
<p class="caption">
Figure 3.1: Before(left) and after(right) batch correction with Limma
</p>
</div>
</div>
<div id="video-demo-of-rima" class="section level2">
<h2><span class="header-section-number">3.5</span> Video demo of RIMA</h2>
<video width="820" height="600" controls>
<source src="videos/preprocess.mp4" type="video/mp4">
</video>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Differential.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
