<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Immune repertoire | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Immune repertoire | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Immune repertoire | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  



<meta name="date" content="2021-05-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Differential.html"/>
<link rel="next" href="Infiltration.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this tutorial</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#pre-built-reference-data"><i class="fa fa-check"></i><b>1.2</b> Pre-built reference data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-mapping"><i class="fa fa-check"></i><b>3.1</b> Read Mapping</a><ul>
<li class="chapter" data-level="3.1.1" data-path="Preprocessing.html"><a href="Preprocessing.html#star"><i class="fa fa-check"></i><b>3.1.1</b> STAR</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a><ul>
<li class="chapter" data-level="3.2.1" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-metrics"><i class="fa fa-check"></i><b>3.2.1</b> Quality metrics</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="Differential.html"><a href="Differential.html#differential-gene-expression-using-deseq2"><i class="fa fa-check"></i><b>4.1</b> Differential gene expression using DESeq2</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Differential.html"><a href="Differential.html#inputs-of-deseq2"><i class="fa fa-check"></i><b>4.1.1</b> Inputs of DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Differential.html"><a href="Differential.html#different-gene-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Different gene expression analysis</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Differential.html"><a href="Differential.html#running-deseq2-with-batch-effect"><i class="fa fa-check"></i><b>4.2.1</b> Running DESeq2 with batch effect</a></li>
<li class="chapter" data-level="4.2.2" data-path="Differential.html"><a href="Differential.html#output-from-deseq2"><i class="fa fa-check"></i><b>4.2.2</b> Output from DESeq2</a></li>
<li class="chapter" data-level="4.2.3" data-path="Differential.html"><a href="Differential.html#volcano-plots"><i class="fa fa-check"></i><b>4.2.3</b> Volcano Plot(s)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Differential.html"><a href="Differential.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4" data-path="Differential.html"><a href="Differential.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a><ul>
<li class="chapter" data-level="5.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4"><i class="fa fa-check"></i><b>5.1</b> TRUST4</a><ul>
<li class="chapter" data-level="5.1.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4-output"><i class="fa fa-check"></i><b>5.1.1</b> TRUST4 output</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="Repertoire.html"><a href="Repertoire.html#bcr-cluster-calculation"><i class="fa fa-check"></i><b>5.2</b> BCR cluster calculation</a></li>
<li class="chapter" data-level="5.3" data-path="Repertoire.html"><a href="Repertoire.html#clonotypes-per-kilo-reads-cpk"><i class="fa fa-check"></i><b>5.3</b> Clonotypes per kilo-reads (CPK)</a></li>
<li class="chapter" data-level="5.4" data-path="Repertoire.html"><a href="Repertoire.html#tcr-and-bcr-entropy-and-colonality"><i class="fa fa-check"></i><b>5.4</b> TCR and BCR entropy and Colonality</a></li>
<li class="chapter" data-level="5.5" data-path="Repertoire.html"><a href="Repertoire.html#fraction-of-reads-mapped-to-tcr-and-bcr"><i class="fa fa-check"></i><b>5.5</b> Fraction of reads mapped to TCR and BCR</a></li>
<li class="chapter" data-level="5.6" data-path="Repertoire.html"><a href="Repertoire.html#somatic-hypermutation-rate-of-bcr"><i class="fa fa-check"></i><b>5.6</b> Somatic hypermutation rate of BCR</a></li>
<li class="chapter" data-level="5.7" data-path="Repertoire.html"><a href="Repertoire.html#ig-isotype-frequency"><i class="fa fa-check"></i><b>5.7</b> Ig isotype frequency</a></li>
<li class="chapter" data-level="5.8" data-path="Repertoire.html"><a href="Repertoire.html#cohort-analysis"><i class="fa fa-check"></i><b>5.8</b> Cohort analysis</a></li>
<li class="chapter" data-level="5.9" data-path="Repertoire.html"><a href="Repertoire.html#video-demo"><i class="fa fa-check"></i><b>5.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-cell-type"><i class="fa fa-check"></i><b>6.7</b> Comparison across cell type</a></li>
<li class="chapter" data-level="6.8" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-sample"><i class="fa fa-check"></i><b>6.8</b> Comparison across sample</a></li>
<li class="chapter" data-level="6.9" data-path="Infiltration.html"><a href="Infiltration.html#video-demo-1"><i class="fa fa-check"></i><b>6.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.1</b> TIDE</a><ul>
<li class="chapter" data-level="7.1.1" data-path="Response.html"><a href="Response.html#cold-and-hot-tumors"><i class="fa fa-check"></i><b>7.1.1</b> Cold and hot tumors</a></li>
<li class="chapter" data-level="7.1.2" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.1.2</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.1.3" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.1.3</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.1.4" data-path="Response.html"><a href="Response.html#normaliztion"><i class="fa fa-check"></i><b>7.1.4</b> Normaliztion</a></li>
<li class="chapter" data-level="7.1.5" data-path="Response.html"><a href="Response.html#running-tide"><i class="fa fa-check"></i><b>7.1.5</b> Running TIDE</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.2</b> Microsatelite instability (MSI)</a></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#response-comparison-analysis-of-biomarkers"><i class="fa fa-check"></i><b>7.3</b> Response comparison analysis of biomarkers</a></li>
<li class="chapter" data-level="7.4" data-path="Response.html"><a href="Response.html#video-demo-2"><i class="fa fa-check"></i><b>7.4</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="HLA.html"><a href="HLA.html"><i class="fa fa-check"></i><b>8</b> HLA Calling</a></li>
<li class="chapter" data-level="9" data-path="Neoantigen.html"><a href="Neoantigen.html"><i class="fa fa-check"></i><b>9</b> Neoantigen Analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="Neoantigen.html"><a href="Neoantigen.html#somatic-mutation"><i class="fa fa-check"></i><b>9.1</b> Somatic mutation</a></li>
<li class="chapter" data-level="9.2" data-path="Neoantigen.html"><a href="Neoantigen.html#neoantigen-identification"><i class="fa fa-check"></i><b>9.2</b> Neoantigen identification</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>10</b> Fusion</a><ul>
<li class="chapter" data-level="10.1" data-path="fusion.html"><a href="fusion.html#fusion-calling-from-chimeric"><i class="fa fa-check"></i><b>10.1</b> Fusion calling from CHIMERIC</a></li>
<li class="chapter" data-level="10.2" data-path="fusion.html"><a href="fusion.html#remove-homolog"><i class="fa fa-check"></i><b>10.2</b> Remove homolog</a></li>
<li class="chapter" data-level="10.3" data-path="fusion.html"><a href="fusion.html#comparison"><i class="fa fa-check"></i><b>10.3</b> Comparison</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="microbiome.html"><a href="microbiome.html"><i class="fa fa-check"></i><b>11</b> Microbiome</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Repertoire" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Immune repertoire</h1>
<p>Once tumors are infiltrated with B cells and T cells, B cells and T cells are potentially activated by BCR and TCR respectively recognizing and binding to tumor-associated antigens. BCR and TCR undergo V(D)J recombination, encoding complementarity-determining region 3 (CDR3) with highly variable peptide sequence binding to the peptide-MHC complex. The V(D)J junctional sequence assembly is a major source of TCR/BCR diversity for recognizing the diverse antigens. Thus, immune repertoire profiling is essential for quantifying B/T cell diversity and clonality, which are essential tumor immune characteristics and key indicators of immunotherapy response. Currently, newly-emerging immune sequencing techniques, such as TCR-seq and BCR-seq, have been designed to sequence B and T cell receptors through quantitative PCR (qPCR). However, the experimental technology is cost-consuming to achieve.</p>
<p>To overcome this difficulty, Tcr Receptor Utilities for Solid Tissue (TRUST) has been developed to infer the immune repertoire from bulk RNA-seq reads, via de novo assembled CDR3 sequences that are not aligned to the host genome. It enables users to overview the immune repertoire of tumors, including CDR3 sequence length, the frequency of various V genes, J genes, and VJ pairs for different chains in the TCR and BCR. For cohort analysis, some basic metrics for TCRs and BCRs across samples could be computed to relate immune repertoire characteristics to specific phenotypes, including the fraction of reads mapped to TCR/BCR, the number of TCR/BCR unique clonotypes of CDR3 sequences, TCR/BCR diversity, and the clonality. Since the activated B cells will undergo somatic hyper-mutation, antibody production, and Immunoglobulin (Ig) class switches during B cell maturation, it is necessary to measure the somatic hypermutation (SHM) rate for BCRs. In addition, TRUST predicts Ig isotypes for BCRs, which allows for quantifying the Ig compositions representing different maturation statuses (IGHM, IGHD, IGHG3, IGHG1, IGHA1, IGHG2, IGHG4, IGHA2, IGHE) and generating an Ig class switches network. Taken together, the immune repertoire analysis quantifies the T/B cell population in both individual and cohort levels, enabling to characterize the dynamic evolution in the TME via the diversity and clonal expansion of tumor-infiltrating T cells or B cells.</p>
<div id="trust4" class="section level2">
<h2><span class="header-section-number">5.1</span> TRUST4</h2>
<p>Example Command:</p>
<pre><code>./run-trust4 -b example.bam -f hg38_bcrtcr.fa --ref human_IMGT+C.fa</code></pre>
<p>The above command utilizes hg3_bcrtcr.fa (fasta file containing the coordinates and sequences of V/D/J/C genes) and human_IMGT+C.fa (V/D/J/C gene reference file from the IMGT database), and produces a simple report (example_report.tsv). This produced report contains a list of assembled clonotypes from example.bam with the following fields: “read_count,frequency(proportion of read_count),CDR3_dna,CDR3_amino_acids,V,D,J,C,consensus_id,consensus_id_full_length”. These fields tell us the abundance, CDR3 sequence (nucleotides and amino acids), gene segment usage, and identifier for each clonotype.</p>
<div id="trust4-output" class="section level3">
<h3><span class="header-section-number">5.1.1</span> TRUST4 output</h3>
</div>
</div>
<div id="bcr-cluster-calculation" class="section level2">
<h2><span class="header-section-number">5.2</span> BCR cluster calculation</h2>
<p>We use one sample for a instance to show how RIMA process the results from TRUST4. We start from a proccessed file that RIMA directly add the sample ID to TRUST4_report.tsv. This step makes it more easier to combine all of individual sample results in following analysis.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="Repertoire.html#cb14-1"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb14-2"><a href="Repertoire.html#cb14-2"></a></span>
<span id="cb14-3"><a href="Repertoire.html#cb14-3"></a><span class="co">#load processed function </span></span>
<span id="cb14-4"><a href="Repertoire.html#cb14-4"></a><span class="kw">source</span>(<span class="st">&quot;TRUST4/trust4_metric_functions.R&quot;</span>)</span>
<span id="cb14-5"><a href="Repertoire.html#cb14-5"></a></span>
<span id="cb14-6"><a href="Repertoire.html#cb14-6"></a><span class="co"># read the individual sample result from RIMA</span></span>
<span id="cb14-7"><a href="Repertoire.html#cb14-7"></a>cdr3 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file =</span> <span class="st">&quot;TRUST4/SRR8281233_cdr3.out.processed.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb14-8"><a href="Repertoire.html#cb14-8"></a></span>
<span id="cb14-9"><a href="Repertoire.html#cb14-9"></a><span class="co">#filter out count of cdr ==0 and add the phenotype info for downstream comparison</span></span>
<span id="cb14-10"><a href="Repertoire.html#cb14-10"></a><span class="co">#This SRR8281233 smaple is Non-responder</span></span>
<span id="cb14-11"><a href="Repertoire.html#cb14-11"></a>phenotype &lt;-<span class="st"> &quot;NR&quot;</span></span>
<span id="cb14-12"><a href="Repertoire.html#cb14-12"></a>cdr3 &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3, count <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb14-13"><a href="Repertoire.html#cb14-13"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">V =</span> <span class="kw">as.character</span>(V), <span class="dt">J =</span> <span class="kw">as.character</span>(J), <span class="dt">C =</span> <span class="kw">as.character</span>(C), <span class="dt">CDR3aa =</span> <span class="kw">as.character</span>(CDR3aa)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb14-14"><a href="Repertoire.html#cb14-14"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">clinic =</span> <span class="kw">as.character</span>(phenotype))</span>
<span id="cb14-15"><a href="Repertoire.html#cb14-15"></a></span>
<span id="cb14-16"><a href="Repertoire.html#cb14-16"></a><span class="kw">head</span>(cdr3)</span></code></pre></div>
<pre><code>##   count  frequency                                           CDR3nt
## 1   125 0.03682734                TGCATGCAAGCTCTACAAACTCCGTACACTTTT
## 2   124 0.03654850                TGTCAGCAGCGTAGCAACTGGCCTTGGACGTTC
## 3    98 0.05410145 TGTGCGATGGGGGATAGTAGTGGCTGGTACCGTCCTCCCGACTCCTGG
## 4    97 0.02847096                   TGTCAACATTACGGTACCTCGTGGACGTTC
## 5    79 0.02320237                TGTCAGCAGCGTAGCAACTGGCCGCTCACTTTC
## 6    78 0.02289418                TGCATGCAGCGTGTGAGTCTTCCCCACACTTTT
##             CDR3aa           V           D        J     C          cid
## 1      CMQALQTPYTF IGKV2-28*01           . IGKJ2*01  IGKC    assemble0
## 2      CQQRSNWPWTF IGKV3-11*01           . IGKJ1*01  IGKC assemble1701
## 3 CAMGDSSGWYRPPDSW IGHV3-33*01 IGHD6-19*01 IGHJ5*01 IGHG1    assemble1
## 4       CQHYGTSWTF IGKV3-20*01           . IGKJ1*01  IGKC assemble1713
## 5      CQQRSNWPLTF IGKV3-11*01           . IGKJ4*01  IGKC assemble1717
## 6      CMQRVSLPHTF IGKV2-40*01           . IGKJ2*01  IGKC assemble1719
##       sample clinic
## 1 SRR8281233     NR
## 2 SRR8281233     NR
## 3 SRR8281233     NR
## 4 SRR8281233     NR
## 5 SRR8281233     NR
## 6 SRR8281233     NR</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="Repertoire.html#cb16-1"></a><span class="co">#determine whether the cdr animo acid is complete or partial</span></span>
<span id="cb16-2"><a href="Repertoire.html#cb16-2"></a>cdr3<span class="op">$</span>is_complete &lt;-<span class="st"> </span><span class="kw">sapply</span>(cdr3<span class="op">$</span>CDR3aa, <span class="cf">function</span>(x) <span class="kw">ifelse</span>(x <span class="op">!=</span><span class="st"> &quot;partial&quot;</span> <span class="op">&amp;&amp;</span><span class="st"> </span>x <span class="op">!=</span><span class="st"> &quot;out_of_frame&quot;</span> <span class="op">&amp;&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;^_&quot;</span>,x) <span class="op">&amp;&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;^</span><span class="ch">\\</span><span class="st">?&quot;</span>, x),<span class="st">&quot;Y&quot;</span>,<span class="st">&quot;N&quot;</span>))</span>
<span id="cb16-3"><a href="Repertoire.html#cb16-3"></a></span>
<span id="cb16-4"><a href="Repertoire.html#cb16-4"></a><span class="co">#exact the TCR and BCR</span></span>
<span id="cb16-5"><a href="Repertoire.html#cb16-5"></a>cdr3.bcr &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3, <span class="kw">grepl</span>(<span class="st">&quot;^IG&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG&quot;</span>,C))</span>
<span id="cb16-6"><a href="Repertoire.html#cb16-6"></a>cdr3.tcr &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3, <span class="kw">grepl</span>(<span class="st">&quot;^TR&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^TR&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^TR&quot;</span>,C))</span>
<span id="cb16-7"><a href="Repertoire.html#cb16-7"></a></span>
<span id="cb16-8"><a href="Repertoire.html#cb16-8"></a><span class="co">#add lib size and clinic traits</span></span>
<span id="cb16-9"><a href="Repertoire.html#cb16-9"></a>cdr3.bcr &lt;-<span class="st"> </span>cdr3.bcr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">lib.size =</span> <span class="kw">sum</span>(count)) </span>
<span id="cb16-10"><a href="Repertoire.html#cb16-10"></a>cdr3.tcr &lt;-<span class="st"> </span>cdr3.tcr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">lib.size =</span> <span class="kw">sum</span>(count)) </span>
<span id="cb16-11"><a href="Repertoire.html#cb16-11"></a></span>
<span id="cb16-12"><a href="Repertoire.html#cb16-12"></a><span class="co">#split BCR into heavy chain and light chain</span></span>
<span id="cb16-13"><a href="Repertoire.html#cb16-13"></a>cdr3.bcr.heavy &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3.bcr, <span class="kw">grepl</span>(<span class="st">&quot;^IGH&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IGH&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IGH&quot;</span>,C))</span>
<span id="cb16-14"><a href="Repertoire.html#cb16-14"></a>cdr3.bcr.light &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3.bcr, <span class="kw">grepl</span>(<span class="st">&quot;^IG[K|L]&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG[K|L]&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG[K|L]&quot;</span>,C))</span>
<span id="cb16-15"><a href="Repertoire.html#cb16-15"></a></span>
<span id="cb16-16"><a href="Repertoire.html#cb16-16"></a><span class="co">#save BCR and TCR info for downsteam use</span></span>
<span id="cb16-17"><a href="Repertoire.html#cb16-17"></a>outdir &lt;-<span class="st"> &quot;TRUST4/individual/&quot;</span></span>
<span id="cb16-18"><a href="Repertoire.html#cb16-18"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb16-19"><a href="Repertoire.html#cb16-19"></a><span class="kw">save</span>(cdr3.bcr.light, <span class="dt">file =</span> <span class="kw">paste</span>(outdir, <span class="st">&quot;TRUST4_BCR_light.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb16-20"><a href="Repertoire.html#cb16-20"></a><span class="kw">save</span>(cdr3.bcr.heavy, <span class="dt">file =</span> <span class="kw">paste</span>(outdir, <span class="st">&quot;TRUST4_BCR_heavy.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb16-21"><a href="Repertoire.html#cb16-21"></a><span class="kw">save</span>(cdr3.tcr, <span class="dt">file =</span> <span class="kw">paste</span>(outdir, <span class="st">&quot;TRUST4_TCR.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb16-22"><a href="Repertoire.html#cb16-22"></a></span>
<span id="cb16-23"><a href="Repertoire.html#cb16-23"></a><span class="co">#BCR clustering </span></span>
<span id="cb16-24"><a href="Repertoire.html#cb16-24"></a><span class="co">#Note that not every sample have BCR cluster</span></span>
<span id="cb16-25"><a href="Repertoire.html#cb16-25"></a>sample_bcr_cluster &lt;-<span class="st"> </span><span class="kw">BuildBCRlineage</span>(<span class="dt">sampleID =</span> sample, <span class="dt">Bdata =</span> cdr3.bcr.heavy, <span class="dt">start=</span><span class="dv">3</span>, <span class="dt">end=</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## use default substitution matrix
## use default substitution matrix
## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="Repertoire.html#cb18-1"></a><span class="kw">save</span>(sample_bcr_cluster,<span class="dt">file =</span> <span class="kw">paste</span>(outdir, sample,<span class="st">&quot;_TRUST4_BCR_heavy_cluster.Rdata&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb18-2"><a href="Repertoire.html#cb18-2"></a></span>
<span id="cb18-3"><a href="Repertoire.html#cb18-3"></a><span class="kw">attributes</span>(sample_bcr_cluster)</span></code></pre></div>
<pre><code>## $names
## [1] &quot;RGSPRFID&quot; &quot;KVPGRRTR&quot; &quot;RNLYYGAY&quot;</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="Repertoire.html#cb20-1"></a><span class="kw">head</span>(sample_bcr_cluster<span class="op">$</span>RGSPRFID)</span></code></pre></div>
<pre><code>## $distMat
##      [,1] [,2] [,3] [,4]
## [1,]    0    1    1    3
## [2,]    0    0    2    4
## [3,]    0    0    0    2
## [4,]    0    0    0    0
## 
## $Sequences
## [1] &quot;TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG&quot;
## [2] &quot;TGTGCGCGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG&quot;
## [3] &quot;TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCTGG&quot;
## [4] &quot;TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCCAG&quot;
## 
## $data
##   count   frequency                                                 CDR3nt
## 1     2 0.001099511 TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG
## 2     2 0.001099511 TGTGCGCGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG
## 3     7 0.003848289 TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCTGG
## 4     2 0.001099511 TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCCAG
##               CDR3aa           V           D        J     C         cid
## 1 CARGSPRFIDYGGSLHFW IGHV4-31*01 IGHD4-23*01 IGHJ4*02 IGHG1 assemble836
## 2 CARGSPRFIDYGGSLHFW IGHV4-31*11 IGHD4-23*01 IGHJ4*02 IGHG1  assemble90
## 3 CARGSPRFIDYGGSFHFW IGHV4-31*11 IGHD4-23*01 IGHJ4*02 IGHG1  assemble90
## 4 CARGSPRFIDYGGSFHFQ IGHV4-31*11 IGHD4-23*01 IGHJ4*02 IGHG1  assemble90
##       sample clinic is_complete lib.size
## 1 SRR8281233     NR           Y     5184
## 2 SRR8281233     NR           Y     5184
## 3 SRR8281233     NR           Y     5184
## 4 SRR8281233     NR           Y     5184</code></pre>
</div>
<div id="clonotypes-per-kilo-reads-cpk" class="section level2">
<h2><span class="header-section-number">5.3</span> Clonotypes per kilo-reads (CPK)</h2>
<p>CPK can be calculated by taking the number of unique CDR3 calls for a chain divided by the total number of reads for that chain, with this result being multiplied by 1000. This is used as a measure of clonotype diversity.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="Repertoire.html#cb22-1"></a><span class="co">#TCR CPK</span></span>
<span id="cb22-2"><a href="Repertoire.html#cb22-2"></a>cpk &lt;-<span class="st"> </span><span class="kw">aggregate</span>(CDR3aa <span class="op">~</span><span class="st"> </span>sample<span class="op">+</span>clinic<span class="op">+</span>lib.size, cdr3.tcr, <span class="cf">function</span>(x) <span class="kw">length</span>(<span class="kw">unique</span>(x))) <span class="op">%&gt;%</span></span>
<span id="cb22-3"><a href="Repertoire.html#cb22-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CPK =</span> <span class="kw">signif</span>(CDR3aa<span class="op">/</span>(lib.size<span class="op">/</span><span class="dv">1000</span>),<span class="dv">4</span>))</span>
<span id="cb22-4"><a href="Repertoire.html#cb22-4"></a>cpk</span></code></pre></div>
<pre><code>##       sample clinic lib.size CDR3aa CPK
## 1 SRR8281233     NR       73     20 274</code></pre>
</div>
<div id="tcr-and-bcr-entropy-and-colonality" class="section level2">
<h2><span class="header-section-number">5.4</span> TCR and BCR entropy and Colonality</h2>
<p>The Shannon entropy index is a measure used for repertoire diversity using clonotype frequencies, which reflects both richness and evenness of the repertoire. This measure informs us of the probability that two random selections from the same repertoire would represent the same clonotype.</p>
<p>Clonality can be measured using normalized entropy over the number of unique clones. It is equivalent to 1 - Pielou’s Evenness, making it inversely proportional to diversity. A higher clonality index indicates an uneven repertoire due to expansion of clones.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="Repertoire.html#cb24-1"></a><span class="co">#BCR clonality and entropy</span></span>
<span id="cb24-2"><a href="Repertoire.html#cb24-2"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb24-3"><a href="Repertoire.html#cb24-3"></a></span>
<span id="cb24-4"><a href="Repertoire.html#cb24-4"></a>single_sample_bcr_clonality &lt;-<span class="st"> </span><span class="kw">getClonality</span>(sample, cdr3.bcr.heavy, <span class="dt">start=</span><span class="dv">3</span>, <span class="dt">end=</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## use default substitution matrix
## use default substitution matrix
## use default substitution matrix
## [1] &quot;SRR8281233&quot;</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="Repertoire.html#cb26-1"></a>single_sample_bcr_clonality[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##               sample            clonality              entropy 
##         &quot;SRR8281233&quot; &quot;0.0899083128267854&quot;   &quot;6.83997594611138&quot;</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="Repertoire.html#cb28-1"></a><span class="co">#TCR clonality and entropy</span></span>
<span id="cb28-2"><a href="Repertoire.html#cb28-2"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb28-3"><a href="Repertoire.html#cb28-3"></a></span>
<span id="cb28-4"><a href="Repertoire.html#cb28-4"></a>single_sample_tcr_clonality &lt;-<span class="st"> </span><span class="kw">getClonalityTCR</span>(sample,cdr3.tcr)</span>
<span id="cb28-5"><a href="Repertoire.html#cb28-5"></a>single_sample_tcr_clonality</span></code></pre></div>
<pre><code>##               sample            clonality              entropy 
##         &quot;SRR8281233&quot; &quot;0.0699534783052342&quot;    &quot;4.0850595412347&quot;</code></pre>
</div>
<div id="fraction-of-reads-mapped-to-tcr-and-bcr" class="section level2">
<h2><span class="header-section-number">5.5</span> Fraction of reads mapped to TCR and BCR</h2>
<p>The fraction of reads mapped to either TCR or BCR is indicated by the V,D,J-gene prefixes of clonotypes. The fraction is composed of the proportion of reads with genes prefixed by “TR” (TCR) and genes prefixed by “IG” (BCR).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="Repertoire.html#cb30-1"></a><span class="co">#exact the mapped reads info from alignment index file </span></span>
<span id="cb30-2"><a href="Repertoire.html#cb30-2"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb30-3"><a href="Repertoire.html#cb30-3"></a></span>
<span id="cb30-4"><a href="Repertoire.html#cb30-4"></a>file &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;TRUST4/SRR8281233.sorted.bam.stat.txt&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb30-5"><a href="Repertoire.html#cb30-5"></a>mapped.reads &lt;-<span class="st"> </span>file[<span class="st">&quot;reads mapped:&quot;</span>,<span class="st">&quot;V2&quot;</span>]</span>
<span id="cb30-6"><a href="Repertoire.html#cb30-6"></a>individual.stats &lt;-<span class="st"> </span><span class="kw">cbind.data.frame</span>(<span class="dt">sample =</span> sample, <span class="dt">map.reads =</span> mapped.reads)</span>
<span id="cb30-7"><a href="Repertoire.html#cb30-7"></a></span>
<span id="cb30-8"><a href="Repertoire.html#cb30-8"></a><span class="co">#---------fraction of BCR reads------------------</span></span>
<span id="cb30-9"><a href="Repertoire.html#cb30-9"></a><span class="co">##extract library size</span></span>
<span id="cb30-10"><a href="Repertoire.html#cb30-10"></a>lib.size &lt;-<span class="st"> </span>cdr3.bcr.heavy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span></span>
<span id="cb30-11"><a href="Repertoire.html#cb30-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">lib =</span> <span class="kw">mean</span>(lib.size))</span>
<span id="cb30-12"><a href="Repertoire.html#cb30-12"></a></span>
<span id="cb30-13"><a href="Repertoire.html#cb30-13"></a><span class="co">##combine stats and library size</span></span>
<span id="cb30-14"><a href="Repertoire.html#cb30-14"></a>bcr.lib.reads &lt;-<span class="st"> </span><span class="kw">merge</span>(individual.stats,lib.size,<span class="dt">by =</span> <span class="st">&quot;sample&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-15"><a href="Repertoire.html#cb30-15"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Infil =</span> <span class="kw">signif</span>(<span class="kw">as.numeric</span>(lib)<span class="op">/</span><span class="kw">as.numeric</span>(map.reads),<span class="dv">4</span>))</span>
<span id="cb30-16"><a href="Repertoire.html#cb30-16"></a></span>
<span id="cb30-17"><a href="Repertoire.html#cb30-17"></a><span class="co">#------------fraction of TCR reads-----------------</span></span>
<span id="cb30-18"><a href="Repertoire.html#cb30-18"></a><span class="co">##extract library size</span></span>
<span id="cb30-19"><a href="Repertoire.html#cb30-19"></a>lib.size &lt;-<span class="st"> </span>cdr3.tcr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span></span>
<span id="cb30-20"><a href="Repertoire.html#cb30-20"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">lib =</span> <span class="kw">mean</span>(lib.size)) </span>
<span id="cb30-21"><a href="Repertoire.html#cb30-21"></a><span class="co">##combine stats and library size</span></span>
<span id="cb30-22"><a href="Repertoire.html#cb30-22"></a>tcr.lib.reads &lt;-<span class="st"> </span><span class="kw">merge</span>(individual.stats,lib.size,<span class="dt">by =</span> <span class="st">&quot;sample&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-23"><a href="Repertoire.html#cb30-23"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Infil =</span> <span class="kw">signif</span>(<span class="kw">as.numeric</span>(lib)<span class="op">/</span><span class="kw">as.numeric</span>(map.reads),<span class="dv">4</span>))</span>
<span id="cb30-24"><a href="Repertoire.html#cb30-24"></a></span>
<span id="cb30-25"><a href="Repertoire.html#cb30-25"></a>combined &lt;-<span class="st"> </span><span class="kw">rbind</span>(bcr.lib.reads, tcr.lib.reads) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Type =</span> <span class="kw">c</span>(<span class="st">&quot;BCR&quot;</span>, <span class="st">&quot;TCR&quot;</span>))</span>
<span id="cb30-26"><a href="Repertoire.html#cb30-26"></a>combined</span></code></pre></div>
<pre><code>##       sample map.reads  lib     Infil Type
## 1 SRR8281233  59972094 5184 8.644e-05  BCR
## 2 SRR8281233  59972094   73 1.217e-06  TCR</code></pre>
</div>
<div id="somatic-hypermutation-rate-of-bcr" class="section level2">
<h2><span class="header-section-number">5.6</span> Somatic hypermutation rate of BCR</h2>
<p>A somatic hypermutation (SHM) is a cellular mechanism of the immune system when adapting to new foreign elements and occurs with a point mutation in the variable regions of Ig genes. TRUST4 collapses reads with SHMs into the same contig by representing a contig as the consensus of the assembled reads. The consensus indicates that a specific nucleotide sequence has the highest weight among the assembled reads.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="Repertoire.html#cb32-1"></a>SHM.ratio &lt;-<span class="st"> </span><span class="kw">getSHMratio</span>(sample_bcr_cluster)</span></code></pre></div>
<pre><code>## [1] 54
## [1] 2
## [1] 48
## [1] 3
## [1] 51
## [1] 1</code></pre>
</div>
<div id="ig-isotype-frequency" class="section level2">
<h2><span class="header-section-number">5.7</span> Ig isotype frequency</h2>
<p>The immunoglobulin heavy (IGH) locus includes the constant heavy (IGHC) gene segment. This gene segment includes different isotypes (IGHA1, IGHA2, IGHD, IGHE, IGHG1, IGHG2, IGHG3, IGHG4, IGHM) and the proportion of the abundance of these different segments indicates the isotype frequency. The isotype frequency can be found by isolating reads with genes prefixed “IGH” and comparing the occurrence of different isotypes in the C_gene column.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="Repertoire.html#cb34-1"></a><span class="co">#This SRR8281233 smaple is Non-responder</span></span>
<span id="cb34-2"><a href="Repertoire.html#cb34-2"></a>phenotype &lt;-<span class="st"> &quot;NR&quot;</span></span>
<span id="cb34-3"><a href="Repertoire.html#cb34-3"></a></span>
<span id="cb34-4"><a href="Repertoire.html#cb34-4"></a>st.Ig &lt;-<span class="st"> </span>cdr3.bcr.heavy <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb34-5"><a href="Repertoire.html#cb34-5"></a><span class="st">  </span><span class="kw">group_by</span>(clinic,sample) <span class="op">%&gt;%</span></span>
<span id="cb34-6"><a href="Repertoire.html#cb34-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">est_clonal_exp_norm =</span> frequency<span class="op">/</span><span class="kw">sum</span>(frequency)) <span class="op">%&gt;%</span><span class="st">   </span><span class="co">#as.numeric(sample.clones[filename,2])</span></span>
<span id="cb34-7"><a href="Repertoire.html#cb34-7"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(C <span class="op">!=</span><span class="st"> &quot;*&quot;</span> <span class="op">&amp;</span><span class="st"> </span>C <span class="op">!=</span><span class="st">&quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb34-8"><a href="Repertoire.html#cb34-8"></a><span class="st">  </span><span class="kw">group_by</span>(sample, C) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb34-9"><a href="Repertoire.html#cb34-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">Num.Ig =</span> <span class="kw">sum</span>(est_clonal_exp_norm)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">clinic =</span> phenotype)</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;sample&#39;. You can override using the `.groups` argument.</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="Repertoire.html#cb36-1"></a>st.Ig</span></code></pre></div>
<pre><code>## # A tibble: 8 x 4
## # Groups:   sample [1]
##   sample     C      Num.Ig clinic
##   &lt;chr&gt;      &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt; 
## 1 SRR8281233 IGHA1 0.134   NR    
## 2 SRR8281233 IGHA2 0.0682  NR    
## 3 SRR8281233 IGHD  0.00385 NR    
## 4 SRR8281233 IGHG1 0.542   NR    
## 5 SRR8281233 IGHG2 0.0704  NR    
## 6 SRR8281233 IGHG3 0.0280  NR    
## 7 SRR8281233 IGHG4 0.0126  NR    
## 8 SRR8281233 IGHM  0.0407  NR</code></pre>
</div>
<div id="cohort-analysis" class="section level2">
<h2><span class="header-section-number">5.8</span> Cohort analysis</h2>
<p>We show the basic immune repertoire-related metrics above using a single sample. Now we combine all of samples in the demo dataset to do some comparison.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="Repertoire.html#cb38-1"></a><span class="co">#load the pre-processed results that contain all samples </span></span>
<span id="cb38-2"><a href="Repertoire.html#cb38-2"></a><span class="co">#These results would be automatically generated after running RIMA immune repertoire module</span></span>
<span id="cb38-3"><a href="Repertoire.html#cb38-3"></a></span>
<span id="cb38-4"><a href="Repertoire.html#cb38-4"></a>inputdir &lt;-<span class="st"> &quot;TRUST4/Cohort/&quot;</span></span>
<span id="cb38-5"><a href="Repertoire.html#cb38-5"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_cluster.Rdata&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-6"><a href="Repertoire.html#cb38-6"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_clonality.Rdata&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-7"><a href="Repertoire.html#cb38-7"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_SHMRatio.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-8"><a href="Repertoire.html#cb38-8"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_lib_reads_Infil.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-9"><a href="Repertoire.html#cb38-9"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_Ig_CS.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-10"><a href="Repertoire.html#cb38-10"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_lib_reads_Infil.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-11"><a href="Repertoire.html#cb38-11"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-12"><a href="Repertoire.html#cb38-12"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_TCR_lib_reads_Infil.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-13"><a href="Repertoire.html#cb38-13"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_TCR.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-14"><a href="Repertoire.html#cb38-14"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_TCR_clonality.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb38-15"><a href="Repertoire.html#cb38-15"></a></span>
<span id="cb38-16"><a href="Repertoire.html#cb38-16"></a></span>
<span id="cb38-17"><a href="Repertoire.html#cb38-17"></a><span class="co">#call the ploting function</span></span>
<span id="cb38-18"><a href="Repertoire.html#cb38-18"></a><span class="kw">source</span>(<span class="st">&quot;TRUST4/trust4_plot.R&quot;</span>)</span>
<span id="cb38-19"><a href="Repertoire.html#cb38-19"></a>meta &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;metasheet.csv&quot;</span>)</span>
<span id="cb38-20"><a href="Repertoire.html#cb38-20"></a>p &lt;-<span class="st"> </span><span class="kw">Trust4_plot</span>(<span class="dt">phenotype =</span> <span class="st">&quot;Responder&quot;</span>, <span class="dt">metasheet =</span> meta)</span>
<span id="cb38-21"><a href="Repertoire.html#cb38-21"></a></span>
<span id="cb38-22"><a href="Repertoire.html#cb38-22"></a>p</span></code></pre></div>
<p><img src="RIMA-tutorial_files/figure-html/unnamed-chunk-19-1.png" width="576" /></p>
</div>
<div id="video-demo" class="section level2">
<h2><span class="header-section-number">5.9</span> Video demo</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Differential.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Infiltration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
