<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Immune Response | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Immune Response | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Immune Response | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  



<meta name="date" content="2021-06-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Infiltration.html"/>
<link rel="next" href="HLA.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html"><i class="fa fa-check"></i><b>2</b> How to run RIMA</a><ul>
<li class="chapter" data-level="2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#installation"><i class="fa fa-check"></i><b>2.1</b> Installation</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#download-pre-built-reference"><i class="fa fa-check"></i><b>2.2</b> Download pre-built reference</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#prepare-input-files"><i class="fa fa-check"></i><b>2.3</b> Prepare input files</a></li>
<li class="chapter" data-level="2.4" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#execute-rima"><i class="fa fa-check"></i><b>2.4</b> Execute RIMA</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#output"><i class="fa fa-check"></i><b>2.5</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-mapping"><i class="fa fa-check"></i><b>3.1</b> Read Mapping</a><ul>
<li class="chapter" data-level="3.1.1" data-path="Preprocessing.html"><a href="Preprocessing.html#star"><i class="fa fa-check"></i><b>3.1.1</b> STAR</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a><ul>
<li class="chapter" data-level="3.2.1" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-metrics"><i class="fa fa-check"></i><b>3.2.1</b> Quality metrics</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="Differential.html"><a href="Differential.html#differential-gene-expression-using-deseq2"><i class="fa fa-check"></i><b>4.1</b> Differential gene expression using DESeq2</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Differential.html"><a href="Differential.html#inputs-of-deseq2"><i class="fa fa-check"></i><b>4.1.1</b> Inputs of DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Differential.html"><a href="Differential.html#different-gene-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Different gene expression analysis</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Differential.html"><a href="Differential.html#running-deseq2-with-batch-effect"><i class="fa fa-check"></i><b>4.2.1</b> Running DESeq2 with batch effect</a></li>
<li class="chapter" data-level="4.2.2" data-path="Differential.html"><a href="Differential.html#output-from-deseq2"><i class="fa fa-check"></i><b>4.2.2</b> Output from DESeq2</a></li>
<li class="chapter" data-level="4.2.3" data-path="Differential.html"><a href="Differential.html#volcano-plots"><i class="fa fa-check"></i><b>4.2.3</b> Volcano Plot(s)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Differential.html"><a href="Differential.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4" data-path="Differential.html"><a href="Differential.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a><ul>
<li class="chapter" data-level="5.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4"><i class="fa fa-check"></i><b>5.1</b> TRUST4</a><ul>
<li class="chapter" data-level="5.1.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4-output"><i class="fa fa-check"></i><b>5.1.1</b> TRUST4 output</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="Repertoire.html"><a href="Repertoire.html#bcr-cluster-calculation"><i class="fa fa-check"></i><b>5.2</b> BCR cluster calculation</a></li>
<li class="chapter" data-level="5.3" data-path="Repertoire.html"><a href="Repertoire.html#clonotypes-per-kilo-reads-cpk"><i class="fa fa-check"></i><b>5.3</b> Clonotypes per kilo-reads (CPK)</a></li>
<li class="chapter" data-level="5.4" data-path="Repertoire.html"><a href="Repertoire.html#tcr-and-bcr-entropy-and-colonality"><i class="fa fa-check"></i><b>5.4</b> TCR and BCR entropy and Colonality</a></li>
<li class="chapter" data-level="5.5" data-path="Repertoire.html"><a href="Repertoire.html#fraction-of-reads-mapped-to-tcr-and-bcr"><i class="fa fa-check"></i><b>5.5</b> Fraction of reads mapped to TCR and BCR</a></li>
<li class="chapter" data-level="5.6" data-path="Repertoire.html"><a href="Repertoire.html#somatic-hypermutation-rate-of-bcr"><i class="fa fa-check"></i><b>5.6</b> Somatic hypermutation rate of BCR</a></li>
<li class="chapter" data-level="5.7" data-path="Repertoire.html"><a href="Repertoire.html#ig-isotype-frequency"><i class="fa fa-check"></i><b>5.7</b> Ig isotype frequency</a></li>
<li class="chapter" data-level="5.8" data-path="Repertoire.html"><a href="Repertoire.html#cohort-analysis"><i class="fa fa-check"></i><b>5.8</b> Cohort analysis</a></li>
<li class="chapter" data-level="5.9" data-path="Repertoire.html"><a href="Repertoire.html#video-demo"><i class="fa fa-check"></i><b>5.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-cell-type"><i class="fa fa-check"></i><b>6.7</b> Comparison across cell type</a></li>
<li class="chapter" data-level="6.8" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-sample"><i class="fa fa-check"></i><b>6.8</b> Comparison across sample</a></li>
<li class="chapter" data-level="6.9" data-path="Infiltration.html"><a href="Infiltration.html#video-demo-1"><i class="fa fa-check"></i><b>6.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.1</b> TIDE</a><ul>
<li class="chapter" data-level="7.1.1" data-path="Response.html"><a href="Response.html#cold-and-hot-tumors"><i class="fa fa-check"></i><b>7.1.1</b> Cold and hot tumors</a></li>
<li class="chapter" data-level="7.1.2" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.1.2</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.1.3" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.1.3</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.1.4" data-path="Response.html"><a href="Response.html#normaliztion"><i class="fa fa-check"></i><b>7.1.4</b> Normaliztion</a></li>
<li class="chapter" data-level="7.1.5" data-path="Response.html"><a href="Response.html#running-tide"><i class="fa fa-check"></i><b>7.1.5</b> Running TIDE</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.2</b> Microsatelite instability (MSI)</a></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#response-comparison-analysis-of-biomarkers"><i class="fa fa-check"></i><b>7.3</b> Response comparison analysis of biomarkers</a></li>
<li class="chapter" data-level="7.4" data-path="Response.html"><a href="Response.html#video-demo-2"><i class="fa fa-check"></i><b>7.4</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="HLA.html"><a href="HLA.html"><i class="fa fa-check"></i><b>8</b> HLA Calling</a></li>
<li class="chapter" data-level="9" data-path="Neoantigen.html"><a href="Neoantigen.html"><i class="fa fa-check"></i><b>9</b> Neoantigen Analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="Neoantigen.html"><a href="Neoantigen.html#somatic-mutation"><i class="fa fa-check"></i><b>9.1</b> Somatic mutation</a></li>
<li class="chapter" data-level="9.2" data-path="Neoantigen.html"><a href="Neoantigen.html#neoantigen-identification"><i class="fa fa-check"></i><b>9.2</b> Neoantigen identification</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>10</b> Fusion</a><ul>
<li class="chapter" data-level="10.1" data-path="fusion.html"><a href="fusion.html#fusion-calling-from-chimeric"><i class="fa fa-check"></i><b>10.1</b> Fusion calling from CHIMERIC</a></li>
<li class="chapter" data-level="10.2" data-path="fusion.html"><a href="fusion.html#remove-homolog"><i class="fa fa-check"></i><b>10.2</b> Remove homolog</a></li>
<li class="chapter" data-level="10.3" data-path="fusion.html"><a href="fusion.html#comparison"><i class="fa fa-check"></i><b>10.3</b> Comparison</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="microbiome.html"><a href="microbiome.html"><i class="fa fa-check"></i><b>11</b> Microbiome</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Response" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Immune Response</h1>
<p>Cancer immunotherapy utilizing immune checkpoint blockade (ICB) has gained excellent progression in treating advanced-stage patients. However, only a small fraction of patients achieved a response to the immunotherapy due to the complexity and heterogeneity of the TME. An increasing number of biomarkers have widely been used to estimate the clinical effectiveness of immunotherapy, such as ICB-related gene signatures (PD1, PDL1, CTLA4), tumor-infiltrating lymphocytes, tumor mutation burden, neoantigens, microsatellite instability (MSI), and serum markers. Nevertheless, there is an urgent need to build a method integrating the multiple dynamic factors to characterize TME and predict immunotherapy response. Currently, two major approaches have been effectively utilized in immunotherapy response prediction. One is models constructed using transcriptome expression profiles of immune checkpoint or T-cell activity, and the other one is artificial intelligence-based algorithms using imaging characteristics of lesions. For tumor RNA-seq data, <a href="http://tide.dfci.harvard.edu/">TIDE</a> models tumor immune evasion by combining gene signatures estimating T cell dysfunction and exclusion to predict ICB response, which may provide critical insights to guide clinical treatment. This computational framework computes the TIDE score for tumors and separates them into responders and non-responders. In detail, TIDE provides T cell dysfunction score, T cell exclusion score, cytotoxic T lymphocyte score, and scores of cell types restricting T cell infiltration in TME, including cancer-associated fibroblasts (CAF), myeloid-derived suppressor cells (MDSC), and M2 macrophages. These metrics scores help the biologists understand the TME status or T cell functions of interest. In this case, tumor RNA-seq profiling leverages the power of gene signature expression for quantifying the dynamic TME status, providing guidance of immunotherapy response prediction for clinicians.</p>
<div id="tide" class="section level2">
<h2><span class="header-section-number">7.1</span> TIDE</h2>
<p>TIDE is a transcriptome biomarker database of ICB response by modeling tumor immune
dysfunction and exclusion. The statistical model of TIDE was trained on clinical tumor profiles without ICB treatments since the immune evasion mechanisms in treatment naïve tumors are also likely to influence patient response to immunotherapies. The TIDE model has been applied to evaluate T cell dysfunction and exclusion signatures across over 33K samples in 188 tumor cohorts from well-curated databases, including TCGA, METABRIC, and PRECOG, as well as our in-house collections.<br />
</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/WsBSi_sH-_g" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p><br />
</p>
<p>A command line TIDE is available at <a href="https://github.com/liulab-dfci/TIDEpy" class="uri">https://github.com/liulab-dfci/TIDEpy</a>.</p>
<div id="cold-and-hot-tumors" class="section level3">
<h3><span class="header-section-number">7.1.1</span> Cold and hot tumors</h3>
<p>TIDE estimate the cytotoxic T Lymphocyte (CTL) level in tumor from the average expression of CD8A, CD8B, GZM, GZMB and PRF1 from treatment naïve tumors. ‘Hot tumors’ have above-average CTL values among all samples, while ‘Cold tumors’ have CTL values below average. TIDE score is the combination of T cell dysfunction estimated from hot tumors and T cell exclusion estimated from cold tumors.<br />
</p>
</div>
<div id="t-cell-dysfunction" class="section level3">
<h3><span class="header-section-number">7.1.2</span> T cell dysfunction</h3>
<p>For hot tumors with high CTL values, the T cell dysfunction is modeled from Cox-PH repression in TIDE:<br />
<span class="math display">\[
Harzard = a*CTL+b*P+d*CTL*P
\]</span>
Where the variable CTL represents the CTLs levels, P represents the status of candidate genes and the coefficient d reflects the effect of interaction between CTL and candidate gene P on dealth harzard. The T cell dysfunction score of individual tumor is calcaulted as:
<span class="math display">\[
Dysfunction = \frac{d}{StdErr(d)}
\]</span></p>
</div>
<div id="t-cell-exclusion" class="section level3">
<h3><span class="header-section-number">7.1.3</span> T cell exclusion</h3>
<p>For hot tumors with high CTL values, TIDE calculates T cell exclusion score to estimate the immune escape. The T cell exlclusion score is the average expression profile from cancer-associated fibroblasts (CAFs), myeloid-derived suppressor cells (MDSCs) and tumor-associated macrophages (TAMs), which are reported to restrict T cell infiltration in tumor.<br />
</p>
</div>
<div id="normaliztion" class="section level3">
<h3><span class="header-section-number">7.1.4</span> Normaliztion</h3>
<p>The input for TIDE is a normalized TPM matrix, the steps for normalization is:<br />
</p>
<ol style="list-style-type: decimal">
<li>Do the log2(TPM+1) transformation.<br />
</li>
<li>Subtract the average across your samples.<br />
</li>
</ol>
</div>
<div id="running-tide" class="section level3">
<h3><span class="header-section-number">7.1.5</span> Running TIDE</h3>
<p>TIDE supports cancer type specific analysis of melanoma and non-small cell lung cancer (NSCLC), our data is from glioblastoma multiforme (GBM), so we set -c as other cancer type and force TIDE to normalized the TPM matrix.<br />
</p>
<p>To run TIDE:</p>
<pre><code>tidepy -o TIDE_score.txt -c Other --force_cormalization tpm_convertID.txt
</code></pre>
<p>Example outputs of TIDE:<br />
</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="Response.html#cb64-1"></a>tide_res &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;TIDE_score.txt&quot;</span>,<span class="dt">header=</span><span class="ot">TRUE</span>,<span class="dt">sep=</span><span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>)</span>
<span id="cb64-2"><a href="Response.html#cb64-2"></a><span class="kw">print</span>(tide_res)</span></code></pre></div>
<pre><code>##             X No.benefits Responder       TIDE        IFNG    MSI.Score
## 1  SRR8281219       False     False  1.6807050  0.05776435 0.9171423549
## 2  SRR8281236       False     False  1.4737267 -0.44182240 0.0891168795
## 3  SRR8281251       False     False  0.3144625  0.72752075 0.9280729250
## 4  SRR8281233       False      True -0.5123185  0.34811417 0.5993703233
## 5  SRR8281244       False      True -0.9853986  0.21431441 0.9799730558
## 6  SRR8281238       False      True -1.9248032 -0.90589129 0.0009766364
## 7  SRR8281230        True     False  1.1296130 -1.26863334 0.2538437260
## 8  SRR8281245       False     False  0.4289897  3.88767936 0.0017826161
## 9  SRR8281250       False     False  0.3857392 -1.24755461 0.9386137813
## 10 SRR8281226       False     False  0.1014286 -1.21158487 0.9948723196
## 11 SRR8281218       False      True -0.6018754 -1.00643540 0.9824508826
## 12 SRR8281243       False      True -0.6690983  0.84652886 0.0098136280
##           CD274        CD8 CTL.flag Dysfunction  Exclusion        MDSC
## 1   1.614004697  1.0782170    False -0.80734665  1.6807050  0.12475420
## 2  -0.571334617 -1.9530623    False -0.35635988  1.4737267  0.11729077
## 3  -0.000579565 -1.3454258    False -0.37740657  0.3144625  0.10977895
## 4  -0.411058820 -0.6337873    False  0.53196445 -0.5123185 -0.14295622
## 5  -0.606725356  3.4653282    False  0.08913789 -0.9853986 -0.10578784
## 6  -0.024306339 -0.6112697    False  0.92111671 -1.9248032 -0.11623220
## 7   0.770622520 -1.5651339    False -0.37601870  1.1296130  0.09720250
## 8   1.057713700  5.4054676     True  0.42898972 -0.5965523 -0.04806820
## 9  -1.316139124 -1.0777690    False -0.23234131  0.3857392  0.05575237
## 10 -0.867420745 -1.6865379    False -0.80196436  0.1014286 -0.01248780
## 11 -1.080325919 -0.2808955    False  0.39883468 -0.6018754 -0.05455741
## 12  1.435549568 -0.7951312    False  0.70257668 -0.6690983 -0.05677437
##            CAF       TAM.M2
## 1   0.15396889 -0.026495315
## 2   0.07706755  0.026260679
## 3  -0.08909806  0.018044264
## 4   0.05095308  0.014947245
## 5  -0.04049768 -0.009330085
## 6  -0.14906967 -0.015234391
## 7   0.06391605  0.003882855
## 8   0.03353989 -0.073846662
## 9  -0.03536659  0.035777532
## 10  0.02156957  0.002648255
## 11 -0.08670749  0.057716177
## 12 -0.01752071 -0.019989212</code></pre>
</div>
</div>
<div id="microsatelite-instability-msi" class="section level2">
<h2><span class="header-section-number">7.2</span> Microsatelite instability (MSI)</h2>
<p>Microsatelite instability (MSI) is the condition of genetic hypermutability (predisposition to mutation) that results from impaired DNA mismatch repair (MMR). Previous study shows that the MSI is highly associated with tumor mutation burden and closly related to immunothrapy response. Many tools MANTIS, MSIsensor, mSINGS, MSIsensor-pro are available to calcualte MSI score form RNA-seq data. in our RIMA pipeline, we applied <a href="https://github.com/niu-lab/msisensor2">MSIsensor2</a> to estimate MSI score, which allows tumor-only BAM input.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="Response.html#cb66-1"></a><span class="co">#Run MSIsensor2 </span></span>
<span id="cb66-2"><a href="Response.html#cb66-2"></a>msisensor2 msi <span class="op">-</span>M models_hg38 <span class="op">-</span>t tumor.bam <span class="op">-</span>o tumor</span>
<span id="cb66-3"><a href="Response.html#cb66-3"></a></span>
<span id="cb66-4"><a href="Response.html#cb66-4"></a><span class="co">## Output of MSIsensor2</span></span>
<span id="cb66-5"><a href="Response.html#cb66-5"></a>Total_Number_of_Sites   Number_of_Somatic_Sites %</span>
<span id="cb66-6"><a href="Response.html#cb66-6"></a><span class="dv">156</span>                     <span class="dv">26</span>                     <span class="fl">16.67</span></span></code></pre></div>
<p>Example outputs of combined MSI:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="Response.html#cb67-1"></a>msi_res &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;msi_score.txt&quot;</span>,<span class="dt">header=</span><span class="ot">FALSE</span>,<span class="dt">sep=</span><span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>)</span>
<span id="cb67-2"><a href="Response.html#cb67-2"></a><span class="kw">print</span>(msi_res)</span></code></pre></div>
<pre><code>##            V1    V2
## 1  SRR8281218 20.34
## 2  SRR8281219 13.39
## 3  SRR8281226 28.00
## 4  SRR8281238 22.11
## 5  SRR8281236 17.69
## 6  SRR8281230 16.10
## 7  SRR8281233 17.97
## 8  SRR8281244 25.71
## 9  SRR8281245 30.43
## 10 SRR8281243 24.11
## 11 SRR8281251 23.44
## 12 SRR8281250 24.59</code></pre>
</div>
<div id="response-comparison-analysis-of-biomarkers" class="section level2">
<h2><span class="header-section-number">7.3</span> Response comparison analysis of biomarkers</h2>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="Response.html#cb69-1"></a><span class="co">#read the metasheet</span></span>
<span id="cb69-2"><a href="Response.html#cb69-2"></a>metafile &lt;-<span class="st"> &quot;metasheet.csv&quot;</span></span>
<span id="cb69-3"><a href="Response.html#cb69-3"></a>tide_res &lt;-<span class="st"> &quot;TIDE_score.txt&quot;</span></span>
<span id="cb69-4"><a href="Response.html#cb69-4"></a>msi_res &lt;-<span class="st"> &quot;msi_score.txt&quot;</span></span>
<span id="cb69-5"><a href="Response.html#cb69-5"></a><span class="co">#load ploting function</span></span>
<span id="cb69-6"><a href="Response.html#cb69-6"></a><span class="kw">source</span>(<span class="st">&quot;Tideploting.R&quot;</span>)</span>
<span id="cb69-7"><a href="Response.html#cb69-7"></a></span>
<span id="cb69-8"><a href="Response.html#cb69-8"></a><span class="kw">cmpr_biomk</span>(<span class="dt">msi_path =</span> msi_res, <span class="dt">tide_path =</span> tide_res, <span class="dt">meta_path =</span> metafile, <span class="dt">phenotype =</span> <span class="st">&quot;Responder&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-32"></span>
<img src="images/response.png" alt="Comparision of biomarkers between R and NR"  />
<p class="caption">
Figure 7.1: Comparision of biomarkers between R and NR
</p>
</div>
</div>
<div id="video-demo-2" class="section level2">
<h2><span class="header-section-number">7.4</span> Video demo</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Infiltration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="HLA.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
